<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Alignment using Rbowtie2 — bowtie_align • CLIPflexR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Alignment using Rbowtie2 — bowtie_align"><meta property="og:description" content="Alignment using Rbowtie2"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CLIPflexR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.21</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Processing_to_matrix.html">Processing reads, mapping, and building a count matrix</a>
    </li>
    <li>
      <a href="../articles/StandardandBrdU_Processing_CTK.html">CTK: Standard/BrdU-CLIP data analysis</a>
    </li>
    <li>
      <a href="../articles/chimera_vignette.html">Chimera processing</a>
    </li>
    <li>
      <a href="../articles/installCliPR.html">Using and installing the CTK pipeline using CLIPflexR</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/kathrynrozengagnon/CLIPflexR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Alignment using Rbowtie2</h1>
    <small class="dont-index">Source: <a href="https://github.com/kathrynrozengagnon/CLIPflexR/blob/HEAD/R/Functions.R" class="external-link"><code>R/Functions.R</code></a></small>
    <div class="hidden name"><code>bowtie_align.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Alignment using Rbowtie2</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">bowtie_align</span><span class="op">(</span></span>
<span>  <span class="va">fq</span>,</span>
<span>  <span class="va">index</span>,</span>
<span>  outbam <span class="op">=</span> <span class="st">"fq"</span>,</span>
<span>  inputFormat <span class="op">=</span> <span class="st">"fasta"</span>,</span>
<span>  keepSAM <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  mode <span class="op">=</span> <span class="st">"genome_map"</span>,</span>
<span>  maxMismatches <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  seedSubString <span class="op">=</span> <span class="fl">18</span>,</span>
<span>  threads <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  report_k <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  keep_all <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  soft_clip <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  additional_Args <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>fq</dt>
<dd><p>path to file to process (fasta or fastq).</p></dd>


<dt>index</dt>
<dd><p>bowtie2 index name without extension.</p></dd>


<dt>outbam</dt>
<dd><p>output bam name, not required if mode is set. If mode is set to "NULL", specify "fq" (default - name of file to process/reads) or "index" (name of index)</p></dd>


<dt>inputFormat</dt>
<dd><p>Format of reads, "fastq" or "fasta" (default).</p></dd>


<dt>keepSAM</dt>
<dd><p>keep bowtie2 SAM output file, TRUE (default) or FALSE.</p></dd>


<dt>mode</dt>
<dd><p>mapping mode, "genome_map" (default - 1 mismatch in seed alignment, seed substring of 18) or "reverse_map" (for mapping short sequences to reads, i.e. miRNAs or chimera analysis, 0 mismatches and a seed substring of 18); set mode=NULL to custom set bowtie2 options using arguments below or additional arguments.</p></dd>


<dt>maxMismatches</dt>
<dd><p>max mismatches in seed alignment, (default is 0).</p></dd>


<dt>seedSubString</dt>
<dd><p>length of seed substrings (default is 18).</p></dd>


<dt>threads</dt>
<dd><p>number of threads to use (default is 1).</p></dd>


<dt>report_k</dt>
<dd><p>number of alignments to report, default is NULL (the best alignment is reported), set to integer to specify.</p></dd>


<dt>keep_all</dt>
<dd><p>report all alignments, default is NULL (the best alignment is reported), set to TRUE to report all.</p></dd>


<dt>soft_clip</dt>
<dd><p>allow soft clipping, default is NULL (no soft clipping), set to TRUE to to soft clip.</p></dd>


<dt>additional_Args</dt>
<dd><p>any additional mapping arguments not set above, default is NULL, can be set by specifying a single character string with spaces between arguments (see example below); run "Rbowtie2::bowtie2_usage()" to see all options; please note that due to a Rbowtie2 bug, the "--no-1mm-upfront" is not available.</p></dd>


<dt>overwrite</dt>
<dd><p>overwrite existing output files, TRUE or FALSE (default).</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>path to sorted BAM.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Kathryn Rozen-Gagnon</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">testFasta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/hg19Small.fa"</span>,package<span class="op">=</span><span class="st">"CLIPflexR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">myIndex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="bowtie2_index.html">bowtie2_index</a></span><span class="op">(</span><span class="va">testFasta</span>, overwrite<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">testFQ</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/Fox3_Std_small.fq.gz"</span>,package<span class="op">=</span><span class="st">"CLIPflexR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">FqFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="decompress.html">decompress</a></span><span class="op">(</span><span class="va">testFQ</span>,overwrite<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">FqFile_FF</span> <span class="op">&lt;-</span> <span class="fu"><a href="ctk_fastqFilter.html">ctk_fastqFilter</a></span><span class="op">(</span><span class="va">FqFile</span>,qsFilter<span class="op">=</span><span class="st">"mean:0-29:20"</span>,verbose<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fastq_filter.pl command is /Users/runner/Library/r-miniconda/envs/CLIPflexR_0.1.21/bin/ctk/fastq_filter.pl</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fastq_filter.pl arguments are /Users/runner/Library/r-miniconda/envs/CLIPflexR_0.1.21/bin/ctk/fastq_filter.pl  -if sanger  -of fastq  -f mean:0-29:20  /Users/runner/work/_temp/Library/CLIPflexR/extdata/Fox3_Std_small.fq  /Users/runner/work/_temp/Library/CLIPflexR/extdata/FF_Fox3_Std_small.fq </span>
<span class="r-in"><span><span class="va">FqFile_clipped</span> <span class="op">&lt;-</span> <span class="fu"><a href="fastx_clipper.html">fastx_clipper</a></span><span class="op">(</span><span class="va">FqFile_FF</span>,length<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">FqFile_QF</span> <span class="op">&lt;-</span> <span class="fu"><a href="fastq_quality_trimmer.html">fastq_quality_trimmer</a></span><span class="op">(</span><span class="va">FqFile_clipped</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">FqFile_Col</span> <span class="op">&lt;-</span> <span class="fu"><a href="ctk_fastq2collapse.html">ctk_fastq2collapse</a></span><span class="op">(</span><span class="va">FqFile_QF</span>,verbose<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fastq_filter.pl command is /Users/runner/Library/r-miniconda/envs/CLIPflexR_0.1.21/bin/ctk/fastq2collapse.pl</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fastq_filter.pl arguments are /Users/runner/Library/r-miniconda/envs/CLIPflexR_0.1.21/bin/ctk/fastq2collapse.pl  /Users/runner/work/_temp/Library/CLIPflexR/extdata/QT_FF_Fox3_Std_small_clip.fq  /Users/runner/work/_temp/Library/CLIPflexR/extdata/Collapsed_QT_FF_Fox3_Std_small_clip.fq </span>
<span class="r-in"><span><span class="va">FqFile_ColStrip</span> <span class="op">&lt;-</span> <span class="fu"><a href="ctk_stripBarcode.html">ctk_stripBarcode</a></span><span class="op">(</span><span class="va">FqFile_Col</span>,linkerlength<span class="op">=</span><span class="fl">5</span>, inputFormat<span class="op">=</span><span class="st">"fastq"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">##map reads to genome</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu">bowtie_align</span><span class="op">(</span><span class="va">FqFile_ColStrip</span>,<span class="va">myIndex</span>,</span></span>
<span class="r-in"><span>inputFormat<span class="op">=</span><span class="st">"fastq"</span>, overwrite<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/Users/runner/work/_temp/Library/CLIPflexR/extdata/Collapsed_QT_FF_Fox3_Std_small_clip_rm5.bam"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">##map reads to genome, custom mode</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu">bowtie_align</span><span class="op">(</span><span class="va">FqFile_ColStrip</span>,<span class="va">myIndex</span>, mode<span class="op">=</span><span class="cn">NULL</span>, </span></span>
<span class="r-in"><span>soft_clip<span class="op">=</span><span class="cn">TRUE</span>, additional_Args<span class="op">=</span><span class="st">"--mp 15 -R 4"</span>, inputFormat<span class="op">=</span><span class="st">"fastq"</span>, overwrite<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/Users/runner/work/_temp/Library/CLIPflexR/extdata/Collapsed_QT_FF_Fox3_Std_small_clip_rm5.bam"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">##map miRNAs to reads</span></span></span>
<span class="r-in"><span><span class="va">miRNAs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/hsa_mature.fa"</span>,package<span class="op">=</span><span class="st">"CLIPflexR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">FaFile_Fa</span> <span class="op">&lt;-</span> <span class="fu"><a href="fastq_to_fasta.html">fastx_qtoa</a></span><span class="op">(</span><span class="va">FqFile_ColStrip</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">readIndex</span> <span class="op">&lt;-</span> <span class="fu"><a href="bowtie2_index.html">bowtie2_index</a></span><span class="op">(</span><span class="va">FaFile_Fa</span>, overwrite<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu">bowtie_align</span><span class="op">(</span><span class="va">miRNAs</span>,<span class="va">readIndex</span>, mode<span class="op">=</span><span class="st">"reverse_map"</span>,overwrite<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/Users/runner/work/_temp/Library/CLIPflexR/extdata/Collapsed_QT_FF_Fox3_Std_small_clip_rm5.bam"</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Kathryn Rozen-Gagnon, Thomas Carroll, Ji-Dung Luo, Wei Wang, Troels Scheel.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

