<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Map unprocessed or processed reads to genome and count small RNAs by location — ranges_count • CLIPflexR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Map unprocessed or processed reads to genome and count small RNAs by location — ranges_count" />
<meta property="og:description" content="Map unprocessed or processed reads to genome and count small RNAs by location" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CLIPflexR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.20</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Processing_to_matrix.html">Processing reads, mapping, and building a count matrix</a>
    </li>
    <li>
      <a href="../articles/StandardandBrdU_Processing_CTK.html">CTK: Standard/BrdU-CLIP data analysis</a>
    </li>
    <li>
      <a href="../articles/chimera_vignette.html">Chimera processing</a>
    </li>
    <li>
      <a href="../articles/installCliPR.html">Using and installing the CTK pipeline using CLIPflexR</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/kathrynrozengagnon/CLIPflexR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Map unprocessed or processed reads to genome and count small RNAs by location</h1>
    <small class="dont-index">Source: <a href='https://github.com/kathrynrozengagnon/CLIPflexR/blob/master/R/Functions.R'><code>R/Functions.R</code></a></small>
    <div class="hidden name"><code>Ranges_count.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Map unprocessed or processed reads to genome and count small RNAs by location</p>
    </div>

    <pre class="usage"><span class='fu'>ranges_count</span><span class='op'>(</span>
  <span class='va'>fastas</span>,
  <span class='va'>miRNA_ranges</span>,
  <span class='va'>genomeIndex</span>,
  linkers <span class='op'>=</span> <span class='cn'>NULL</span>,
  length_max <span class='op'>=</span> <span class='cn'>NULL</span>,
  length_min <span class='op'>=</span> <span class='cn'>NULL</span>,
  mode <span class='op'>=</span> <span class='cn'>NULL</span>,
  maxMismatches <span class='op'>=</span> <span class='fl'>0</span>,
  threads <span class='op'>=</span> <span class='fl'>1</span>,
  report_k <span class='op'>=</span> <span class='cn'>NULL</span>,
  keep_all <span class='op'>=</span> <span class='cn'>NULL</span>,
  soft_clip <span class='op'>=</span> <span class='cn'>NULL</span>,
  additional_Args <span class='op'>=</span> <span class='st'>"--score-min C,0,0"</span>,
  seedSubString <span class='op'>=</span> <span class='fl'>18</span>,
  overwrite <span class='op'>=</span> <span class='cn'>FALSE</span>,
  bpparam <span class='op'>=</span> <span class='cn'>NULL</span>,
  verbose <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>fastas</th>
      <td><p>path to FASTA files to process.</p></td>
    </tr>
    <tr>
      <th>miRNA_ranges</th>
      <td><p>BED file containing annotated miRNA or other small RNA genomic ranges.</p></td>
    </tr>
    <tr>
      <th>genomeIndex</th>
      <td><p>path to genome index.</p></td>
    </tr>
    <tr>
      <th>linkers</th>
      <td><p>contaminating sequences to remove, default is NULL, set to character string to specify.</p></td>
    </tr>
    <tr>
      <th>length_max</th>
      <td><p>maximum length required during fasta processing, default is NULL, set to integer to specify.</p></td>
    </tr>
    <tr>
      <th>length_min</th>
      <td><p>minimum length required during fasta processing, default is NULL, set to integer to specify.</p></td>
    </tr>
    <tr>
      <th>mode</th>
      <td><p>mapping mode, default is NULL to allow custom mapping with default arguments set below (0 mismatches in seed alignment, seed substring length 18, disallow alignments with mismatches); see bowtie_align for other mapping modes.</p></td>
    </tr>
    <tr>
      <th>maxMismatches</th>
      <td><p>max mismatches in seed alignment (default is 0).</p></td>
    </tr>
    <tr>
      <th>threads</th>
      <td><p>number of threads to use (default is 1).</p></td>
    </tr>
    <tr>
      <th>report_k</th>
      <td><p>number of alignments to report, default is NULL (the best alignment is reported), set to integer to specify.</p></td>
    </tr>
    <tr>
      <th>keep_all</th>
      <td><p>set to TRUE to report all alignments, default is NULL (the best alignment is reported).</p></td>
    </tr>
    <tr>
      <th>soft_clip</th>
      <td><p>set to TRUE to allow soft clipping, default is no soft clipping.</p></td>
    </tr>
    <tr>
      <th>additional_Args</th>
      <td><p>any additional mapping arguments, default is "--score-min C,0,0" to disallow alignments with mismatches; run "Rbowtie2::bowtie2_usage()" to see all options; please note that due to a Rbowtie2 bug, the "--no-1mm-upfront" is not available.</p></td>
    </tr>
    <tr>
      <th>seedSubString</th>
      <td><p>length of seed substrings (default is 18).</p></td>
    </tr>
    <tr>
      <th>overwrite</th>
      <td><p>overwrite existing output files, TRUE or FALSE (default).</p></td>
    </tr>
    <tr>
      <th>bpparam</th>
      <td><p>TRUE or FALSE (default)</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>print messages, TRUE or FALSE (default).</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>path to count matrix.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Kathryn Rozen-Gagnon</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>testFastq</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata/SRR1742056.fastq.gz"</span>,package<span class='op'>=</span><span class='st'>"CLIPflexR"</span><span class='op'>)</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Kathryn Rozen-Gagnon, Thomas Carroll, Ji-Dung Luo, Wei Wang, Troels Scheel.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


