<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Standard/BrdU-CLIP data analysis using CTK • CLIPflexR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Standard/BrdU-CLIP data analysis using CTK">
<meta property="og:description" content="CLIPflexR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CLIPflexR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.18</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Processing_to_matrix.html">Processing reads, mapping, and building a count matrix</a>
    </li>
    <li>
      <a href="../articles/StandardandBrdU_Processing_CTK.html">Standard/BrdU-CLIP data analysis using CTK</a>
    </li>
    <li>
      <a href="../articles/installCliPR.html">Using and installing the CTK pipeline using CLIPflexR</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Standard/BrdU-CLIP data analysis using CTK</h1>
                        <h4 class="author">Kathryn Rozen-Gagnon, Ji-Dung Luo and Thomas Carroll</h4>
            
      
      
      <div class="hidden name"><code>StandardandBrdU_Processing_CTK.Rmd</code></div>

    </div>

    
    
<div id="download-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-data" class="anchor"></a>Download Data</h2>
<p>Download data from <a href="https://www.ncbi.nlm.nih.gov/sra">SRA</a>. <em>SRR110753</em> was selected as an example in this vignette.</p>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="no">SRAdb</span>)
<span class="no">mySRA</span> <span class="kw">&lt;-</span> <span class="st">"SRAmetadb.sqlite"</span>
<span class="kw">if</span>(!<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="no">mySRA</span>)){
  <span class="no">mySRA</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SRAdb/man/getSRAdbFile.html">getSRAdbFile</a></span>()
}
<span class="no">sra_con</span> <span class="kw">&lt;-</span> <span class="fu">dbConnect</span>(<span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>), <span class="no">mySRA</span> )</pre></body></html></div>
<div class="sourceCode"><html><body><pre class="r"><span class="no">Fox3_Std</span> <span class="kw">&lt;-</span> <span class="st">"SRR1107535.fastq.gz"</span>
<span class="kw">if</span>(!<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="no">Fox3_Std</span>)){
  <span class="fu"><a href="https://rdrr.io/pkg/SRAdb/man/getFASTQfile.html">getFASTQfile</a></span>(<span class="st">"SRR1107535"</span>,
               <span class="kw">sra_con</span> <span class="kw">=</span> <span class="no">sra_con</span>,
               <span class="kw">destDir</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span>())
}</pre></body></html></div>
</div>
<div id="process-data" class="section level2">
<h2 class="hasAnchor">
<a href="#process-data" class="anchor"></a>Process data</h2>
<div id="quality-filter-reads-wrapping-ctks-fastq_filter-pl" class="section level3">
<h3 class="hasAnchor">
<a href="#quality-filter-reads-wrapping-ctks-fastq_filter-pl" class="anchor"></a>Quality filter reads <strong>(wrapping CTK’s fastq_filter.pl)</strong>
</h3>
<p>Filtering low quality read using R-wrapped <strong>fastq_filter.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#fastq_filter.pl">CTK</a>. Use fastq/gzipped fast files as input. <strong>outFile</strong>: fastq file after filtering low quality reads; <strong>qsFilter</strong>: filtering criteria, “mean:0-29:20” means the average quality of the first 30 bases must be equal or greater than 20.<br><a href="docs/ref/ctk_fastqFilter.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">Fox3_Std</span> <span class="kw">&lt;-</span> <span class="st">"SRR1107535.fastq.gz"</span>
<span class="no">Fox3_Std_filtered</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/ctk_fastqFilter.html">ctk_fastqFilter</a></span>(<span class="no">Fox3_Std</span>,
                                     <span class="kw">outFile</span> <span class="kw">=</span> <span class="st">"SRR1107535_FF.fastq"</span>,
                                     <span class="kw">qsFilter</span><span class="kw">=</span><span class="st">"mean:0-29:20"</span>)</pre></body></html></div>
</div>
<div id="remove-3-linkers-wrapping-fastxs-fastx_clipper" class="section level3">
<h3 class="hasAnchor">
<a href="#remove-3-linkers-wrapping-fastxs-fastx_clipper" class="anchor"></a>Remove 3’ linkers <strong>(wrapping FASTX’s fastx_clipper)</strong>
</h3>
<p>Trimming 3’-end adapter sequences using R-wrapped <strong>fastx_clipper</strong> function from <a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html#fastx_clipper_usage">FASTX_toolkit</a>. <strong>adaptor</strong>: CLIP this sequence from the 3’-end of reads, “GTGTCAGTCACTTCCAGCGG” (default); <strong>length</strong>: minimum read length, only keep reads with length equal or greater then this threshold, 18 (default). <a href="fastx_clipper.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">Fox3_Std_filtered</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fastx_clipper.html">fastx_clipper</a></span>(<span class="no">Fox3_Std_filtered</span>,<span class="kw">outFile</span> <span class="kw">=</span> <span class="st">"SRR1107535_clipped.fastq"</span>,<span class="kw">adaptor</span> <span class="kw">=</span> <span class="st">"GTGTCAGTCACTTCCAGCGG"</span>,<span class="kw">length</span> <span class="kw">=</span><span class="fl">20</span>)</pre></body></html></div>
</div>
<div id="trim-low-quality-nucleotides-and-short-reads-wrapping-fastxs-fastq_quality_trimmer" class="section level3">
<h3 class="hasAnchor">
<a href="#trim-low-quality-nucleotides-and-short-reads-wrapping-fastxs-fastq_quality_trimmer" class="anchor"></a>Trim low quality nucleotides and short reads <strong>(wrapping FASTX’s fastq_quality_trimmer)</strong>
</h3>
<p>Discard reads with low quality bases and and short reads using R-wrapped <strong>fastq_quality_trimmer</strong> function from <a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html">FASTX-toolkit</a>. <strong>outfile</strong>: file name and path of the fastq files after filtering; <strong>qualityThreshold</strong>: the minimum quality threshold. <a href="fastq_quality_trimmer.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">Fox3_Std_filtered</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fastq_quality_trimmer.html">fastq_quality_trimmer</a></span>(<span class="no">Fox3_Std_filtered</span>,
                                           <span class="kw">outFile</span> <span class="kw">=</span> <span class="st">"SRR1107535_clipFiltered.fastq.gz"</span>,<span class="kw">qualityThreshold</span> <span class="kw">=</span> <span class="fl">5</span>,<span class="kw">minimumLength</span> <span class="kw">=</span> <span class="fl">20</span> )</pre></body></html></div>
</div>
<div id="collapse-exact-ctk_fastq2collapse-wrapping-ctkss-ctk_fastq2collapse-pl" class="section level3">
<h3 class="hasAnchor">
<a href="#collapse-exact-ctk_fastq2collapse-wrapping-ctkss-ctk_fastq2collapse-pl" class="anchor"></a>Collapse exact ctk_fastq2collapse <strong>(wrapping CTKs’s ctk_fastq2collapse.pl)</strong>
</h3>
<p>Collapse duplicate sequences into unique reads using R-wrapped <strong>fastq2collapse.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#fastq2collapse.pl">CTK</a>. <a href="ctk_fastq2collapse.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">Fox3_Std_filtered</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/ctk_fastq2collapse.html">ctk_fastq2collapse</a></span>(<span class="no">Fox3_Std_filtered</span>,
                                           <span class="kw">outFile</span> <span class="kw">=</span> <span class="st">"SRR1107535_collapsed.fastq.gz"</span>)</pre></body></html></div>
</div>
<div id="remove-random-barcodes-wrapping-ctks-stripbarcode-pl" class="section level3">
<h3 class="hasAnchor">
<a href="#remove-random-barcodes-wrapping-ctks-stripbarcode-pl" class="anchor"></a>Remove random barcodes <strong>(wrapping CTK’s stripBarcode.pl)</strong>
</h3>
<p>Strip random barcodes with specific length from the reads using R-wrapped <strong>stripBarcode.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#stripBarcode.pl">CTK</a>. <strong>linkerlength</strong>: barcode length; <strong>inputFormat</strong>: format of input sequences, fasta (default) or fastq. <a href="ctk_stripBarcode.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">Fox3_Std_filtered</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/ctk_stripBarcode.html">ctk_stripBarcode</a></span>(<span class="no">Fox3_Std_filtered</span>,<span class="kw">linkerlength</span> <span class="kw">=</span> <span class="fl">5</span>,
                                      <span class="kw">inputFormat</span> <span class="kw">=</span> <span class="st">"fasta"</span>,
                                           <span class="kw">outFile</span> <span class="kw">=</span> <span class="st">"SRR1107535_stripped.fastq.gz"</span>)</pre></body></html></div>
</div>
<div id="make-reference-data-for-mm10" class="section level3">
<h3 class="hasAnchor">
<a href="#make-reference-data-for-mm10" class="anchor"></a>Make reference data for mm10</h3>
<p>Fetch reference sequences from <a href="https://www.bioconductor.org/packages/release/bioc/html/BSgenome.html">BSgenome</a>.</p>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="no">BSgenome.Mmusculus.UCSC.mm10</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="no">magrittr</span>)
<span class="no">listOfSeq</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fu">seqnames</span>(<span class="no">BSgenome.Mmusculus.UCSC.mm10</span>),<span class="kw">function</span>(<span class="no">x</span>)<span class="no">BSgenome.Mmusculus.UCSC.mm10</span><span class="kw">[[</span><span class="no">x</span>]])
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">listOfSeq</span>) <span class="kw">&lt;-</span> <span class="fu">seqnames</span>(<span class="no">BSgenome.Mmusculus.UCSC.mm10</span>)
<span class="no">toWrite</span> <span class="kw">&lt;-</span> <span class="fu">DNAStringSet</span>(<span class="no">listOfSeq</span>[!<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"Un|random|hap"</span>,<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">listOfSeq</span>))])
<span class="fu">writeXStringSet</span>(<span class="no">toWrite</span>,<span class="kw">file</span><span class="kw">=</span><span class="st">"mm10.fa"</span>)</pre></body></html></div>
</div>
<div id="build-a-bowtie2-genome-index-using-rbowtie2-" class="section level3">
<h3 class="hasAnchor">
<a href="#build-a-bowtie2-genome-index-using-rbowtie2-" class="anchor"></a>Build a Bowtie2 genome index using Rbowtie2.</h3>
<p>Build Bowtie2 index of reference genome, mm10 in this vignette. <strong>threads</strong>: number of threads used. <a href="bowtie2_index.html">Index documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="../reference/bowtie2_index.html">bowtie2_index</a></span>(<span class="st">"mm10.fa"</span>,<span class="kw">threads</span> <span class="kw">=</span> <span class="fl">6</span>)</pre></body></html></div>
</div>
<div id="align-to-genome-using-rbowtie2-" class="section level3">
<h3 class="hasAnchor">
<a href="#align-to-genome-using-rbowtie2-" class="anchor"></a>Align to genome using Rbowtie2.</h3>
<p>Align reads to indexed reference genome using Rbowtie2. <strong>format</strong>: format of input sequences, fasta (default) or fastq; <strong>threads</strong>: number of threads used, 1 (default); <strong>maxMismatches</strong>: number of mismatches allowed, 1 (default) or 0; <strong>seedSubString</strong>: length of seed substrings, 18 (default), must be &gt;3 &amp; &lt;32; <strong>report_k</strong>: NULL, look for multiple alignments, report best (default), otherwise report up to n (integer) alignments per read. <a href="bowtie_align.html">Alignment documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">Fox3_Std_filtered</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/decompress.html">decompress</a></span>(<span class="no">Fox3_Std_filtered</span>)
<span class="no">alignedFile</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/bowtie_align.html">bowtie_align</a></span>(<span class="no">Fox3_Std_filtered</span>,
              <span class="st">"mm10"</span>,
              <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">"\\.fastq"</span>,<span class="st">".bam"</span>,<span class="no">Fox3_Std_filtered</span>),
              <span class="kw">format</span><span class="kw">=</span><span class="st">"fastq"</span>,
              <span class="kw">threads</span> <span class="kw">=</span> <span class="fl">6</span>)</pre></body></html></div>
</div>
<div id="parsing-sam-files" class="section level3">
<h3 class="hasAnchor">
<a href="#parsing-sam-files" class="anchor"></a>Parsing SAM files</h3>
<p>Parsing SAM files toget unique mapped reads and mismatches per read using R-wrapped <strong>parseAlignment.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#parseAlignment.pl">CTK</a>. <strong>minLen</strong>: miminal mapping length to report, 18 (default); <strong>mapQual</strong>: Minimum map quality, 1 (default MAPQ), keep only uniquely mapped reads when mapQual&gt;=1. <a href="ctk_parseAlignment.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">parsedAlignement</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/ctk_parseAlignment.html">ctk_parseAlignment</a></span>(<span class="no">alignedFile</span>,
                                       <span class="kw">minLen</span> <span class="kw">=</span> <span class="fl">18</span>,
                                       <span class="kw">mapQual</span> <span class="kw">=</span> <span class="fl">1</span>,
                                       <span class="kw">mutationFile</span> <span class="kw">=</span> <span class="st">"mutations.txt"</span>)</pre></body></html></div>
</div>
<div id="collapse-duplicated-mapped-reads" class="section level3">
<h3 class="hasAnchor">
<a href="#collapse-duplicated-mapped-reads" class="anchor"></a>Collapse duplicated mapped reads</h3>
<p>Collapse CLIP tags generated by parseAlignment using R-wrapped <strong>tag2collapse.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#tag2collapse.pl">CTK</a> of CTK. <strong>bigFile</strong>: set TRUE if input file is big; <strong>keepMaxScore</strong>: keep the tag with the most weight (instead of the longest one) as representative; <strong>keepTagName</strong>: do not change tag name (no extra information); <strong>weight</strong>: consider the weight of each tag; <strong>weightInName</strong>: find weight in name; <strong>randomBarcode</strong>: random barcode exists, no collapse for different barcodes; <strong>em</strong>: EM threshold to infer reliability of each collapsed read (when have random linker, -1 = no EM); <strong>outputSeqError</strong>: output sequencing errors estimated by the EM algorithm. <a href="ctk_tag2collapse.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="co"># -v -big --random-barcode -EM 30 --seq-error-model alignment -weight --weight-in-name --keep-max-score --keep-tag-name $f.tag.bed</span>
<span class="no">tagcollapsed</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/ctk_tag2collapse.html">ctk_tag2collapse</a></span>(<span class="no">parsedAlignement</span>,
                                     <span class="kw">keepMaxScore</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                                     <span class="kw">keepTagName</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                                     <span class="kw">weight</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                                     <span class="kw">bigFile</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                                     <span class="kw">weightInName</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                                     <span class="kw">randomBarcode</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                                     <span class="kw">em</span> <span class="kw">=</span> <span class="fl">30</span>,
                                     <span class="kw">outputSeqError</span> <span class="kw">=</span> <span class="fl">TRUE</span>
                                     )</pre></body></html></div>
</div>
<div id="join-tags-and-mutations" class="section level3">
<h3 class="hasAnchor">
<a href="#join-tags-and-mutations" class="anchor"></a>Join tags and mutations</h3>
<p>Get the mutations in unique tags using R-wrapped <strong>selectRow.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#selectRow.pl">CTK</a> (originally wrapped from Galaxy). <strong>file1</strong>: mutation file; <strong>file2</strong>: tagged bed file; <strong>field1</strong>: query column; <strong>field2</strong>: filter column; <strong>mode</strong>: running mode, set N while only print paired rows. <a href="ctk_joinWrapper.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">tagJoined</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/ctk_joinWrapper.html">ctk_joinWrapper</a></span>(<span class="st">"mutations.txt"</span>,<span class="no">tagcollapsed</span>,<span class="kw">field1</span> <span class="kw">=</span> <span class="fl">4</span>,<span class="kw">field2</span> <span class="kw">=</span> <span class="fl">4</span>,<span class="kw">mode</span> <span class="kw">=</span> <span class="st">"n"</span>,
                                <span class="kw">outFile</span><span class="kw">=</span><span class="st">"Fox3_3_tag.unique.mutation.txt"</span>)</pre></body></html></div>
</div>
<div id="label-bed-file-with-rgb-information" class="section level3">
<h3 class="hasAnchor">
<a href="#label-bed-file-with-rgb-information" class="anchor"></a>label bed file with rgb information</h3>
<p>Add colors to bed file using R-wrapped <strong>bed2rgb.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#bed2rgb.pl">CTK</a>. <strong>filesToRun</strong>: taged bed files; <strong>col</strong>: assign color in r,g,b format <a href="ctk_bed2rgb.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">tagcollapsedRGB</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/ctk_bed2rgb.html">ctk_bed2rgb</a></span>(<span class="kw">filesToRun</span> <span class="kw">=</span> <span class="no">tagcollapsed</span>,<span class="kw">col</span> <span class="kw">=</span> <span class="st">"128,0,0"</span>)</pre></body></html></div>
</div>
<div id="generate-bedgraph" class="section level3">
<h3 class="hasAnchor">
<a href="#generate-bedgraph" class="anchor"></a>Generate bedgraph</h3>
<p>Generate bedgraphs for visualization using R-wrapped <strong>tag2profile.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#tag2profile.pl">CTK</a>. <strong>bigFile</strong>: set TRUE if input file is big; <strong>ss</strong>: separate strand, TRUE (default); <strong>exact</strong>: exact count of each nucleotide, TRUE (default). <a href="ctk_tag2profile.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="co"># -v -big --random-barcode -EM 30 --seq-error-model alignment -weight --weight-in-name --keep-max-score --keep-tag-name $f.tag.bed</span>
<span class="no">tagProfile</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/ctk_tag2profile.html">ctk_tag2profile</a></span>(<span class="kw">filesToRun</span> <span class="kw">=</span> <span class="no">tagcollapsedRGB</span>,<span class="kw">bigFile</span> <span class="kw">=</span> <span class="fl">TRUE</span>,<span class="kw">ss</span> <span class="kw">=</span> <span class="fl">TRUE</span>,<span class="kw">exact</span> <span class="kw">=</span> <span class="fl">TRUE</span>,<span class="kw">outputFormat</span> <span class="kw">=</span> <span class="st">"bedgraph"</span>)</pre></body></html></div>
</div>
<div id="peak-calling-without-statistical-assessment" class="section level3">
<h3 class="hasAnchor">
<a href="#peak-calling-without-statistical-assessment" class="anchor"></a>Peak calling without statistical assessment</h3>
<p>Call peaks using R-wrapped <strong>tag2peak.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#tag2peak.pl">CTK</a> <em>without</em> statistical assessment. <strong>filesToRun</strong>: RGB labeled tagged bed files; <strong>valleySeeking</strong>: find candidate peaks by valley seeking, TRUE (default); <strong>valleyDepth</strong>: depth of valley, if valley seeking, 0.9 (default), between 0.5 and 1; <strong>bigFile</strong>: set TRUE if the input file is big; <strong>ss</strong>: separate strand, TRUE (default). <a href="ctk_tag2peak.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">tagPeaksNoStats</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/ctk_tag2peak.html">ctk_tag2peak</a></span>(<span class="kw">filesToRun</span> <span class="kw">=</span> <span class="no">tagcollapsedRGB</span>,<span class="kw">valleySeeking</span> <span class="kw">=</span> <span class="fl">TRUE</span>,<span class="kw">valleyDepth</span> <span class="kw">=</span> <span class="fl">0.9</span>,<span class="kw">bigFile</span> <span class="kw">=</span> <span class="fl">TRUE</span>,<span class="kw">ss</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
</div>
<div id="peak-calling-with-statistical-assessment" class="section level3">
<h3 class="hasAnchor">
<a href="#peak-calling-with-statistical-assessment" class="anchor"></a>Peak calling with statistical assessment</h3>
<p>Call peaks using R-wrapped <strong>tag2peak.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#tag2peak.pl">CTK</a> <em>with</em> statistical assessment. <strong>filesToRun</strong>: RGB labeled tagged bed files; <strong>valleySeeking</strong>: find candidate peaks by valley seeking, TRUE (default);<strong>valleyDepth</strong>: depth of valley, if valley seeking, 0.9 (default), between 0.5 and 1; <strong>bigFile</strong>: set TRUE if the input file is big; <strong>ss</strong>: separate strand, TRUE (default); <strong>genes</strong>: gene locus in bed format. <a href="ctk_tag2peak.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="no">TxDb.Mmusculus.UCSC.mm10.knownGene</span>)
<span class="fu">export.bed</span>(<span class="fu">genes</span>(<span class="no">TxDb.Mmusculus.UCSC.mm10.knownGene</span>),<span class="st">"genes.bed"</span>)
<span class="no">tagProfile</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/ctk_tag2peak.html">ctk_tag2peak</a></span>(<span class="kw">filesToRun</span> <span class="kw">=</span> <span class="no">tagcollapsedRGB</span>,
                           <span class="kw">outFile</span> <span class="kw">=</span> <span class="st">"TC_SRR1107535_stripped.RGB.peakWithStats.bed"</span>,
                           <span class="kw">outBoundary</span> <span class="kw">=</span><span class="st">"TC_SRR1107535_stripped.RGB.peakBoundary.bed"</span>,
                           <span class="kw">outHalfPH</span> <span class="kw">=</span> <span class="st">"TC_SRR1107535_stripped.RGB.peakHalf.bed"</span>,
                           <span class="kw">valleySeeking</span> <span class="kw">=</span> <span class="fl">TRUE</span>,<span class="kw">valleyDepth</span> <span class="kw">=</span> <span class="fl">0.9</span>,<span class="kw">bigFile</span> <span class="kw">=</span> <span class="fl">TRUE</span>,<span class="kw">ss</span> <span class="kw">=</span> <span class="fl">TRUE</span>,<span class="kw">genes</span> <span class="kw">=</span> <span class="st">"genes.bed"</span>)</pre></body></html></div>
</div>
<div id="retrieve-sequence-under-peaks-using-clipflexr-fetchsequencesforclip" class="section level3">
<h3 class="hasAnchor">
<a href="#retrieve-sequence-under-peaks-using-clipflexr-fetchsequencesforclip" class="anchor"></a>Retrieve sequence under peaks using CLIPflexR <strong>fetchSequencesForClIP</strong>
</h3>
<p><strong>peaks</strong>: tagged bed file or GRanges of peaks; <strong>reSize</strong>: resize peak to this width around peak center, 64 (default); <strong>fasta</strong>: reference sequence <a href="fetchSequencesForClIP.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">myseq</span> <span class="kw">&lt;-</span> <span class="fu">fetchSequencesForClIP</span>(<span class="kw">peaks</span> <span class="kw">=</span> <span class="no">tagProfile</span>,<span class="kw">reSize</span> <span class="kw">=</span> <span class="fl">20</span>,<span class="kw">fasta</span> <span class="kw">=</span> <span class="st">"mm10.fa"</span>,<span class="kw">bedHeader</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
</div>
<div id="retrieve-peaks-associated-with-particular-motifs-using-clipflexr-annotatepeakswithpatterns" class="section level3">
<h3 class="hasAnchor">
<a href="#retrieve-peaks-associated-with-particular-motifs-using-clipflexr-annotatepeakswithpatterns" class="anchor"></a>Retrieve peaks associated with particular motifs using CLIPflexR <strong>annotatePeaksWithPatterns</strong>
</h3>
<p><strong>peaks</strong>: tagged bed file or GRanges of peaks; <strong>reSize</strong>: resize peak to this width around peak center, 64 (default); <strong>fasta</strong>: reference sequence; <strong>pattern</strong>: FASTA of motif sequence(s) <a href="../reference/annotatePeaksWithPatterns.html">CLIPflexR documentation</a></p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">myseq</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/annotatePeaksWithPatterns.html">annotatePeaksWithPatterns</a></span>(<span class="kw">peaks</span> <span class="kw">=</span> <span class="no">tagProfile</span>,<span class="kw">patterns</span> <span class="kw">=</span> <span class="st">"TGCATG"</span>,<span class="kw">resize</span> <span class="kw">=</span> <span class="fl">100</span>,<span class="kw">fasta</span> <span class="kw">=</span> <span class="st">"mm10.fa"</span>,<span class="kw">bedHeader</span><span class="kw">=</span><span class="fl">FALSE</span>)</pre></body></html></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kathryn Rozen-Gagnon, Thomas Carroll, Ji-Dung Luo, Wei Wang, Troels Scheel.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
