<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Processing reads, mapping, and building a count matrix • CLIPflexR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Processing reads, mapping, and building a count matrix">
<meta property="og:description" content="CLIPflexR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CLIPflexR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.18</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Processing_to_matrix.html">Processing reads, mapping, and building a count matrix</a>
    </li>
    <li>
      <a href="../articles/StandardandBrdU_Processing_CTK.html">Standard/BrdU-CLIP data analysis using CTK</a>
    </li>
    <li>
      <a href="../articles/installCliPR.html">Using and installing the CTK pipeline using CLIPflexR</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/kathrynrozengagnon/CLIPflexR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Processing_to_matrix_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Processing reads, mapping, and building a count matrix</h1>
                        <h4 class="author">Kathryn Rozen-Gagnon, Ji-Dung Luo and Thomas Carroll</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/kathrynrozengagnon/CLIPflexR/blob/master/vignettes/Processing_to_matrix.Rmd"><code>vignettes/Processing_to_matrix.Rmd</code></a></small>
      <div class="hidden name"><code>Processing_to_matrix.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="kw">baseDir</span> <span class="op">&lt;-</span> <span class="st">"/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette"</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://stringr.tidyverse.org">stringr</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/kathrynrozengagnon/CLIPflexR">CLIPflexR</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">SummarizedExperiment</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">GenomicRanges</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">rtracklayer</span>)
</pre></div>
<div id="process-standardbrdu-clip-libraries-using-clipflexr" class="section level2">
<h2 class="hasAnchor">
<a href="#process-standardbrdu-clip-libraries-using-clipflexr" class="anchor"></a>Process standard/BrdU-CLIP libraries using CLIPflexR</h2>
<p>This is an alternative to the CTK processing pipeline for CLIP data, and returns a peak matrix with genomic locations of peaks and counts in each peak by sample</p>
</div>
<div id="initial-processing-and-quality-control" class="section level2">
<h2 class="hasAnchor">
<a href="#initial-processing-and-quality-control" class="anchor"></a>Initial processing and quality control</h2>
<div id="decompress-fastq-files" class="section level3">
<h3 class="hasAnchor">
<a href="#decompress-fastq-files" class="anchor"></a>Decompress fastq files</h3>
<p>Use compressed fastq files as input; R-wrapped decompression function using bzip2. <a href="../reference/decompress.html">CLIPflexR documentation</a></p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co">#define the path to your compressed fastq file(s)</span>
<span class="co">#here we have one example fastq.gz generated by standard CLIP (linker ligations; KRG092419_S1_L001_R1_001.fastq.gz)</span>
<span class="co">#and one example fastq.gz generated by BrdU-CLIP (KRG091819_S1_L001_R1_001.fastq.gz)</span>
<span class="kw">testFQ</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span>(<span class="kw">baseDir</span>, pattern = <span class="st">".gz"</span>, full.names = <span class="fl">TRUE</span>)
<span class="kw">testFQ</span>
</pre></div>
<pre><code>## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001.fastq.gz"
## [2] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001.fastq.gz"</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="kw">FqFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">testFQ</span>, <span class="kw">decompress</span>, overwrite=<span class="fl">TRUE</span>)
<span class="kw">FqFile</span>
</pre></div>
<pre><code>## [[1]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001.fastq"
## 
## [[2]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001.fastq"</code></pre>
</div>
<div id="you-can-look-at-the-quality-of-your-data-wrapping-fastxs-quality-stats" class="section level3">
<h3 class="hasAnchor">
<a href="#you-can-look-at-the-quality-of-your-data-wrapping-fastxs-quality-stats" class="anchor"></a>You can look at the quality of your data <strong>(wrapping FASTX’s quality stats)</strong>
</h3>
<p>Get quality statistics from your run using R-wrapped <strong>fastx_quality_stats</strong> <a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html#fastq_statistics_usage">FASTX_toolkit</a> <a href="../reference/fastx_quality_stats.html">CLIPflexR documentation</a></p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="co">#defaults are set for illumina quality encoding, minimum quality score of 20 for 80% of bases</span>
<span class="kw">stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">FqFile</span>, <span class="kw">fastx_quality_stats</span>)
<span class="kw">stats</span>
</pre></div>
<pre><code>## [[1]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001.txt"
## 
## [[2]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001.txt"</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="kw">stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="kw">stats</span>[[<span class="fl">1</span>]],header = <span class="kw">T</span>, sep = <span class="st">"\t"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">stats</span>)
</pre></div>
<pre><code>##   column    count min max       sum  mean Q1 med Q3 IQR lW rW A_Count
## 1      1 22052373  16  36 712698638 32.32 32  33 34   2 29 36 7104877
## 2      2 22052373   2  36 696285989 31.57 32  33 34   2 29 36 6911254
## 3      3 22052373  16  36 717065818 32.52 32  33 34   2 29 36 7053127
## 4      4 22052373  16  36 718409762 32.58 32  33 34   2 29 36 7049046
## 5      5 22052373  16  36 715824000 32.46 32  33 34   2 29 36 7020505
## 6      6 22052373  14  37 775710593 35.18 34  37 37   3 30 37 6997082
##   C_Count G_Count T_Count N_Count Max_count
## 1 4814492 3873005 6259999       0  22052373
## 2 4622185 3607472 6192199  719263  22052373
## 3 5082531 3618036 6298679       0  22052373
## 4 5060538 3650501 6292288       0  22052373
## 5 5121322 3657985 6252561       0  22052373
## 6 5109816 3666739 6278736       0  22052373</code></pre>
</div>
<div id="quality-filter-wrapping-fastxs-quality-filter" class="section level3">
<h3 class="hasAnchor">
<a href="#quality-filter-wrapping-fastxs-quality-filter" class="anchor"></a>Quality filter <strong>(wrapping FastX’s quality filter)</strong>
</h3>
<p>Filtering fastq files using R-wrapped <strong>fastq_quality_filter</strong> from <a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html#fastq_quality_filter_usage">FASTX_toolkit</a>. <strong>qEncoding</strong>: Quality encoding, default = 33 (corresponds to Illumina and Sanger quality scores); <strong>minimumQuality</strong>: minimum quality score to keep, default = 20. Only keep reads with quality score greater than this threshold. <strong>minimumPercentOfRead</strong>: minimum percent of bases that must have [-q] minimumQuality. <a href="../reference/fastq_quality_filter.html">CLIPflexR documentation</a></p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="kw">FqFile_QF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">FqFile</span>, <span class="kw">fastq_quality_filter</span>)
<span class="kw">FqFile_QF</span>
</pre></div>
<pre><code>## [[1]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/QF_KRG091819_S1_L001_R1_001.fastq"
## 
## [[2]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/QF_KRG092419_S1_L001_R1_001.fastq"</code></pre>
</div>
<div id="fastx_collapser-wrapping-fastxs-fastx_collapser" class="section level3">
<h3 class="hasAnchor">
<a href="#fastx_collapser-wrapping-fastxs-fastx_collapser" class="anchor"></a>Fastx_collapser <strong>(wrapping FASTX’s fastx_collapser)</strong>
</h3>
<p>Collapse duplicate sequences into unique reads using R-wrapped <strong>fastx_collapser</strong> from <a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html#fastx_collapser_usage.">FASTX_toolkit</a>. <strong>qEncoding</strong>: Quality encoding, default = 33 (corresponds to Illumina and Sanger quality scores). <a href="../reference/fastx_collapser.html">CLIPflexR documentation</a></p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="kw">coll_FqFile_QF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">FqFile_QF</span>, <span class="kw">fastx_collapser</span>)
<span class="kw">coll_FqFile_QF</span>
</pre></div>
<pre><code>## [[1]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/QF_KRG091819_S1_L001_R1_001_collapse.fasta"
## 
## [[2]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/QF_KRG092419_S1_L001_R1_001_collapse.fasta"</code></pre>
</div>
</div>
<div id="follow-these-steps-to-process-standard-clip-libraries" class="section level2">
<h2 class="hasAnchor">
<a href="#follow-these-steps-to-process-standard-clip-libraries" class="anchor"></a>Follow these steps to process standard CLIP libraries</h2>
<div id="de-multiplex-samples-wrapping-fastxs-barcode-splitter" class="section level3">
<h3 class="hasAnchor">
<a href="#de-multiplex-samples-wrapping-fastxs-barcode-splitter" class="anchor"></a>De-multiplex samples <strong>(wrapping FASTX’s barcode splitter)</strong>
</h3>
<p>De-multiplex samples by indices using R-wrapped <strong>fastx_barcode_splitter</strong> from <a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html#fastx_barcode_splitter_usage">FASTX_toolkit</a>. <strong>mismatches</strong>: number of mismatches allowed in barcode sequence, 0 (default); <strong>bcFile</strong>: barcode file, see example below. <a href="../reference/fastx_barcode_splitter.html">CLIPflexR documentation</a></p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="co">#provide the path to your barcode file,  which must contain one sample name and its corresponding index sequence for  each line, and be tab delimited</span>

<span class="co">#For example:</span>
<span class="co">#Aag2_rIgG_Exp6A CGATG</span>
<span class="co">#Aag2_rIgG_Exp6B TTAGG</span>

<span class="kw">BCfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span>(<span class="kw">baseDir</span>, pattern=<span class="st">"BC"</span>,  full.names = <span class="fl">TRUE</span>)
<span class="kw">BCfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"stats"</span>, <span class="kw">BCfile</span>, value =<span class="kw">T</span>, invert = <span class="kw">T</span>)
<span class="kw">BCfile</span>
</pre></div>
<pre><code>## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_BC.txt"
## [2] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_BC.txt"</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="co">#take standard libraries only</span>
<span class="kw">samples_split_std</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fastx_barcode_splitter.html">fastx_barcode_splitter</a></span>(<span class="kw">coll_FqFile_QF</span>[[<span class="fl">2</span>]], bcFile = <span class="kw">BCfile</span>[[<span class="fl">2</span>]], verbose = <span class="fl">TRUE</span>)
</pre></div>
<pre><code>## fastx_collapser command is cat /Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/QF_KRG092419_S1_L001_R1_001_collapse.fasta | /Users/kathryn/Library/r-miniconda/envs/CLIPflexR_0.1.18/bin/fastx_barcode_splitter.pl --bcfile /Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_BC.txt --bol --mismatches 0 --prefix 'KRG092419_S1_L001_R1_001_collapse_'</code></pre>
<pre><code>## fastx_collapser arguments are  --bcfile /Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_BC.txt--bol--mismatches 0 --prefix 'KRG092419_S1_L001_R1_001_collapse_'</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="co">#files will be written with a prefix appended for the run, and then the sample name given in the barcode file</span>
<span class="co">#split samples will not have an extension but will be in FASTA format</span>
<span class="kw">samples_split_std</span>
</pre></div>
<pre><code>##  [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7A"   
##  [2] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7B"   
##  [3] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago2_Exp7A"   
##  [4] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago2_Exp7B"   
##  [5] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_mIgG_Exp7A"   
##  [6] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_mIgG_Exp7B"   
##  [7] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_rIgG_Exp7A"   
##  [8] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_rIgG_Exp7B"   
##  [9] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_Ago2_Exp7A"
## [10] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_Ago2_Exp7B"
## [11] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_mIgG_Exp7A"
## [12] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_mIgG_Exp7B"
## [13] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_mIgG_Exp7C"
## [14] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_mIgG_Exp7D"</code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="co">#sample_stats will also be written, and contain how many reads were extracted for each sample and the file name for that run ID</span>
<span class="kw">std_samples_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="kw">baseDir</span>, <span class="st">"/KRG092419_BC_stats.txt"</span>), header = <span class="kw">T</span>, sep =  <span class="st">"\t"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">std_samples_stats</span>)
</pre></div>
<pre><code>##           Barcode  Count                                          Location
## 1 Aag2_Ago1_Exp7A 612767 KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7A
## 2 Aag2_Ago1_Exp7B 579807 KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7B
## 3 Aag2_Ago2_Exp7A 894236 KRG092419_S1_L001_R1_001_collapse_Aag2_Ago2_Exp7A
## 4 Aag2_Ago2_Exp7B 701068 KRG092419_S1_L001_R1_001_collapse_Aag2_Ago2_Exp7B
## 5 Aag2_mIgG_Exp7A 310822 KRG092419_S1_L001_R1_001_collapse_Aag2_mIgG_Exp7A
## 6 Aag2_mIgG_Exp7B  46491 KRG092419_S1_L001_R1_001_collapse_Aag2_mIgG_Exp7B</code></pre>
</div>
<div id="now-remove-the-5-index-linker-and-random-barcode-sequences-wrapping-ctks-stripbarcode-pl" class="section level3">
<h3 class="hasAnchor">
<a href="#now-remove-the-5-index-linker-and-random-barcode-sequences-wrapping-ctks-stripbarcode-pl" class="anchor"></a>Now remove the 5’ index, linker, and random barcode sequences <strong>(wrapping CTK’s stripBarcode.pl)</strong>
</h3>
<p>Strip random barcodes/linkers with specific length from reads using R-wrapped <strong>stripBarcode.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#stripBarcode.pl">CTK</a>. <strong>linkerlength</strong>: barcode length; <strong>inputformat</strong>: format of input sequences, fasta (default). <a href="../reference/ctk_stripBarcode.html">CLIPflexR documentation</a></p>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="kw">samples_split_std_rm5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">samples_split_std</span>, <span class="kw">ctk_stripBarcode</span>, linkerlength = <span class="fl">27</span>, inputFormat=<span class="st">"fasta"</span>)
<span class="kw">samples_split_std_rm5</span>
</pre></div>
<pre><code>## [[1]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7A_rm5.fa"
## 
## [[2]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7B_rm5.fa"
## 
## [[3]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago2_Exp7A_rm5.fa"
## 
## [[4]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago2_Exp7B_rm5.fa"
## 
## [[5]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_mIgG_Exp7A_rm5.fa"
## 
## [[6]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_mIgG_Exp7B_rm5.fa"
## 
## [[7]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_rIgG_Exp7A_rm5.fa"
## 
## [[8]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_rIgG_Exp7B_rm5.fa"
## 
## [[9]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_Ago2_Exp7A_rm5.fa"
## 
## [[10]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_Ago2_Exp7B_rm5.fa"
## 
## [[11]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_mIgG_Exp7A_rm5.fa"
## 
## [[12]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_mIgG_Exp7B_rm5.fa"
## 
## [[13]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_mIgG_Exp7C_rm5.fa"
## 
## [[14]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_mIgG_Exp7D_rm5.fa"</code></pre>
</div>
<div id="now-remove-the-3-linker-wrapping-fastxs-clipper" class="section level3">
<h3 class="hasAnchor">
<a href="#now-remove-the-3-linker-wrapping-fastxs-clipper" class="anchor"></a>Now remove the 3’ linker <strong>(wrapping FASTX’s clipper)</strong>
</h3>
<p>Trimming 3’-end adapter sequences using R-wrapped <strong>fastx_clipper</strong> function from <a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html#fastx_clipper_usage">FASTX_toolkit</a>. <strong>adaptor</strong>: clip this sequence from the 3’-end of reads, “GTGTCAGTCACTTCCAGCGG” (default); <strong>length</strong>: minimum read length, only keep reads with length equal or greater then this threshold, 18 (default). <a href="../reference/fastx_clipper.html">CLIPflexR documentation</a></p>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="kw">samples_split_std_rm5_rm3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">samples_split_std_rm5</span>, <span class="kw">fastx_clipper</span>, length = <span class="fl">18</span>, adaptor=<span class="st">"GTGTCAGTCACTTCCAGCGG"</span>)
<span class="kw">samples_split_std_rm5_rm3</span>
</pre></div>
<pre><code>## [[1]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7A_rm5_clip.fa"
## 
## [[2]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7B_rm5_clip.fa"
## 
## [[3]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago2_Exp7A_rm5_clip.fa"
## 
## [[4]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago2_Exp7B_rm5_clip.fa"
## 
## [[5]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_mIgG_Exp7A_rm5_clip.fa"
## 
## [[6]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_mIgG_Exp7B_rm5_clip.fa"
## 
## [[7]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_rIgG_Exp7A_rm5_clip.fa"
## 
## [[8]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_rIgG_Exp7B_rm5_clip.fa"
## 
## [[9]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_Ago2_Exp7A_rm5_clip.fa"
## 
## [[10]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_Ago2_Exp7B_rm5_clip.fa"
## 
## [[11]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_mIgG_Exp7A_rm5_clip.fa"
## 
## [[12]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_mIgG_Exp7B_rm5_clip.fa"
## 
## [[13]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_mIgG_Exp7C_rm5_clip.fa"
## 
## [[14]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_aegypti_mIgG_Exp7D_rm5_clip.fa"</code></pre>
</div>
<div id="these-reads-are-now-ready-to-map-to-your-genome" class="section level3">
<h3 class="hasAnchor">
<a href="#these-reads-are-now-ready-to-map-to-your-genome" class="anchor"></a>These reads are now ready to map to your genome</h3>
</div>
</div>
<div id="follow-these-steps-to-process-brdu-clip-libraries" class="section level2">
<h2 class="hasAnchor">
<a href="#follow-these-steps-to-process-brdu-clip-libraries" class="anchor"></a>Follow these steps to process BrdU-CLIP libraries</h2>
<div id="first-strip-random-barcode-sequences-umi-from-the-5-end-of-reads-wrapping-ctks-stripbarcode-pl" class="section level3">
<h3 class="hasAnchor">
<a href="#first-strip-random-barcode-sequences-umi-from-the-5-end-of-reads-wrapping-ctks-stripbarcode-pl" class="anchor"></a>First strip random barcode sequences (UMI) from the 5’ end of reads <strong>(wrapping CTK’s stripBarcode.pl)</strong>
</h3>
<p>Strip random barcodes/linkers with specific length from reads using R-wrapped <strong>stripBarcode.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#stripBarcode.pl">CTK</a>. <strong>linkerlength</strong>: barcode length; <strong>inputformat</strong>: format of input sequences. (fasta/fastq). <a href="../reference/ctk_stripBarcode.html">CLIPflexR documentation</a></p>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="kw">BrdU_rm5</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ctk_stripBarcode.html">ctk_stripBarcode</a></span>(<span class="kw">coll_FqFile_QF</span>[[<span class="fl">1</span>]], linkerlength = <span class="fl">7</span>, inputFormat=<span class="st">"fasta"</span>)
<span class="kw">BrdU_rm5</span>
</pre></div>
<pre><code>## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/QF_KRG091819_S1_L001_R1_001_collapse_rm5.fasta"</code></pre>
</div>
<div id="de-multiplex-samples-wrapping-fastxs-barcode-splitter-1" class="section level3">
<h3 class="hasAnchor">
<a href="#de-multiplex-samples-wrapping-fastxs-barcode-splitter-1" class="anchor"></a>De-multiplex samples <strong>(wrapping FASTX’s barcode splitter)</strong>
</h3>
<p>De-multiplex samples by indices using R-wrapped <strong>fastx_barcode_splitter</strong> from <a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html#fastx_barcode_splitter_usage">FASTX_toolkit</a>. <strong>mismatches</strong>: number of mismatches allowed in barcode sequence, 0 (default); <strong>bcFile</strong>: barcode file see example above. <a href="../reference/fastx_barcode_splitter.html">CLIPflexR documentation</a></p>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="kw">samples_split_BrdU</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fastx_barcode_splitter.html">fastx_barcode_splitter</a></span>(<span class="kw">BrdU_rm5</span>, bcFile = <span class="kw">BCfile</span>[[<span class="fl">1</span>]], verbose = <span class="fl">TRUE</span>)
</pre></div>
<pre><code>## fastx_collapser command is cat /Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/QF_KRG091819_S1_L001_R1_001_collapse_rm5.fasta | /Users/kathryn/Library/r-miniconda/envs/CLIPflexR_0.1.18/bin/fastx_barcode_splitter.pl --bcfile /Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_BC.txt --bol --mismatches 0 --prefix 'KRG091819_S1_L001_R1_001_collapse_rm5_'</code></pre>
<pre><code>## fastx_collapser arguments are  --bcfile /Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_BC.txt--bol--mismatches 0 --prefix 'KRG091819_S1_L001_R1_001_collapse_rm5_'</code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit">
<span class="co">#files will be written with a prefix appended for the run, and then the sample name given in the barcode file</span>
<span class="co">#split samples will not have an extension but will be in FASTA format</span>
<span class="kw">samples_split_BrdU</span>
</pre></div>
<pre><code>##  [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6A"   
##  [2] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6B"   
##  [3] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago2_Exp6A"   
##  [4] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago2_Exp6B"   
##  [5] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_mIgG_Exp6A"   
##  [6] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_mIgG_Exp6B"   
##  [7] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_rIgG_Exp6A"   
##  [8] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_rIgG_Exp6B"   
##  [9] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_Ago1_Exp6A"
## [10] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_Ago1_Exp6B"
## [11] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_Ago2_Exp6A"
## [12] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_Ago2_Exp6B"
## [13] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_mIgG_Exp6A"
## [14] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_mIgG_Exp6B"
## [15] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_rIgG_Exp6A"
## [16] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_rIgG_Exp6B"</code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit">
<span class="co">#sample_stats will also be written, and contain how many reads were extracted for each sample and the file name for that run ID</span>
<span class="kw">BrdU_samples_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="kw">baseDir</span>, <span class="st">"/KRG091819_BC_stats.txt"</span>), header = <span class="kw">T</span>, sep =  <span class="st">"\t"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">BrdU_samples_stats</span>)
</pre></div>
<pre><code>##           Barcode  Count
## 1 Aag2_Ago1_Exp6A 491943
## 2 Aag2_Ago1_Exp6B 507596
## 3 Aag2_Ago2_Exp6A 673944
## 4 Aag2_Ago2_Exp6B 393337
## 5 Aag2_mIgG_Exp6A   1688
## 6 Aag2_mIgG_Exp6B    926
##                                                Location
## 1 KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6A
## 2 KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6B
## 3 KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago2_Exp6A
## 4 KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago2_Exp6B
## 5 KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_mIgG_Exp6A
## 6 KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_mIgG_Exp6B</code></pre>
</div>
<div id="trim-3-not-g-ddd-and-index-6nt-from-the-5-end-of-reads-wrapping-fastxs-fastx_trimmer" class="section level3">
<h3 class="hasAnchor">
<a href="#trim-3-not-g-ddd-and-index-6nt-from-the-5-end-of-reads-wrapping-fastxs-fastx_trimmer" class="anchor"></a>Trim 3 “not G” (DDD) and index (6nt) from the 5’ end of reads <strong>(wrapping FASTX’s fastx_trimmer)</strong>
</h3>
<p>Trimming 5’-end adapter sequences using R-wrapped <strong>fastx_trimmer</strong> function from <a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html#fastx_trimmer_usage">FASTX_toolkit</a>. <strong>read_start</strong>: read starts at this base, 10 (default); <strong>read_end</strong> read ends at this base, NULL (default) <a href="../reference/fastx_trimmer.html">CLIPflexR documentation</a></p>
<pre><code>## [[1]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6A_trim.fa"
## 
## [[2]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6B_trim.fa"
## 
## [[3]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago2_Exp6A_trim.fa"
## 
## [[4]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago2_Exp6B_trim.fa"
## 
## [[5]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_mIgG_Exp6A_trim.fa"
## 
## [[6]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_mIgG_Exp6B_trim.fa"
## 
## [[7]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_rIgG_Exp6A_trim.fa"
## 
## [[8]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_rIgG_Exp6B_trim.fa"
## 
## [[9]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_Ago1_Exp6A_trim.fa"
## 
## [[10]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_Ago1_Exp6B_trim.fa"
## 
## [[11]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_Ago2_Exp6A_trim.fa"
## 
## [[12]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_Ago2_Exp6B_trim.fa"
## 
## [[13]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_mIgG_Exp6A_trim.fa"
## 
## [[14]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_mIgG_Exp6B_trim.fa"
## 
## [[15]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_rIgG_Exp6A_trim.fa"
## 
## [[16]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_rIgG_Exp6B_trim.fa"</code></pre>
</div>
<div id="fastx_clipper-wrapping-fastxs-fastx_clipper" class="section level3">
<h3 class="hasAnchor">
<a href="#fastx_clipper-wrapping-fastxs-fastx_clipper" class="anchor"></a>Fastx_clipper <strong>(wrapping FASTX’s fastx_clipper)</strong>
</h3>
<p>Trimming 3’-end adapter sequences using R-wrapped <strong>fastx_clipper</strong> function from <a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html#fastx_clipper_usage">FASTX_toolkit</a>. <strong>adaptor</strong>: clip this sequence from the 3’-end of reads, “GTGTCAGTCACTTCCAGCGG” (default); <strong>length</strong>: minimum read length, only keep reads with length equal or greater then this threshold, 18 (default). <a href="../reference/fastx_clipper.html">CLIPflexR documentation</a></p>
<pre><code>## [[1]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6A_trim_clip.fa"
## 
## [[2]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6B_trim_clip.fa"
## 
## [[3]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago2_Exp6A_trim_clip.fa"
## 
## [[4]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago2_Exp6B_trim_clip.fa"
## 
## [[5]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_mIgG_Exp6A_trim_clip.fa"
## 
## [[6]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_mIgG_Exp6B_trim_clip.fa"
## 
## [[7]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_rIgG_Exp6A_trim_clip.fa"
## 
## [[8]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_rIgG_Exp6B_trim_clip.fa"
## 
## [[9]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_Ago1_Exp6A_trim_clip.fa"
## 
## [[10]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_Ago1_Exp6B_trim_clip.fa"
## 
## [[11]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_Ago2_Exp6A_trim_clip.fa"
## 
## [[12]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_Ago2_Exp6B_trim_clip.fa"
## 
## [[13]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_mIgG_Exp6A_trim_clip.fa"
## 
## [[14]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_mIgG_Exp6B_trim_clip.fa"
## 
## [[15]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_rIgG_Exp6A_trim_clip.fa"
## 
## [[16]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_aegypti_rIgG_Exp6B_trim_clip.fa"</code></pre>
<p>These reads are now ready to map to your genome</p>
</div>
</div>
<div id="build-a-bowtie2-genome-index-and-align-to-genome-using-rbowtie2-" class="section level2">
<h2 class="hasAnchor">
<a href="#build-a-bowtie2-genome-index-and-align-to-genome-using-rbowtie2-" class="anchor"></a>Build a Bowtie2 genome index and align to genome using Rbowtie2.</h2>
<p>Build Bowtie2 index of reference genome, AaegL5 in this vignette. <a href="../reference/bowtie2_index.html">Index documentation</a>; see also <a href="StandardandBrdU_Processing_CTK_editJD.html">CTK processing vignette</a></p>
<p>Align reads to genome by bowtie. <strong>format</strong>: format of input sequences, fasta (default) or fastq; <strong>threads</strong>: number of threads used, 1 (default); <strong>maxMismatches</strong>: number of mismatches allowed, 1 (default) or 0; <strong>seedSubString</strong>: length of seed substrings, 18 (default), must be &gt;3 &amp; &lt;32; <strong>report_k</strong>: NULL, look for multiple alignments, report best (default), otherwise report up to n (integer) alignments per read. <a href="../reference/bowtie_align.html">Alignment documentation</a></p>
<div class="sourceCode" id="cb38"><pre class="downlit">
<span class="co">#you will need to get the genome fasta of your organism and input the path to this fasta</span>
<span class="kw">genomeFasta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="kw">baseDir</span>, <span class="st">"/Aedes-aegypti-LVP_AGWG_CHROMOSOMES_AaegL5.fa"</span>)

<span class="co">#make index; this step may take a long time and is therefore commented out but shown to illustrate how to execute the function</span>
<span class="co">#building the index only needs to be done the first time you map to a given genome</span>

<span class="co">#genomeIndex &lt;- bowtie2_index(genomeFasta)</span>

<span class="co">#If your genome index is already made, you can simply point to the index</span>
<span class="kw">genomeIndex</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="kw">baseDir</span>, <span class="st">"/Aedes-aegypti-LVP_AGWG_CHROMOSOMES_AaegL5"</span>)

<span class="co">#Here I am selecting just a few samples for illustrative purposes. To run all your samples, you would do:</span>
<span class="co">#tomap &lt;- c(BrdU_split_rm5_rm3, samples_split_std_rm5_rm3)</span>
<span class="kw">tomap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">BrdU_split_rm5_rm3</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], <span class="kw">samples_split_std_rm5_rm3</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>])  <span class="co"># all I am doing here is taking samples 1-4 for each set of processed files (BrdU and standard) for mapping</span>
<span class="kw">tomap</span>
</pre></div>
<pre><code>## [[1]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6A_trim_clip.fa"
## 
## [[2]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6B_trim_clip.fa"
## 
## [[3]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7A_rm5_clip.fa"
## 
## [[4]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7B_rm5_clip.fa"</code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit">
<span class="co">#map files to genome with default parameters: maxMismatches=1,seedSubString=18,threads=1,report_k=NULL</span>
<span class="kw">bams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">tomap</span>, <span class="kw">bowtie_align</span>, <span class="kw">genomeIndex</span>)
<span class="kw">bams</span>
</pre></div>
<pre><code>## [[1]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/Sorted_KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6A_trim_clip.fa.bam"
## 
## [[2]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/Sorted_KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6B_trim_clip.fa.bam"
## 
## [[3]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/Sorted_KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7A_rm5_clip.fa.bam"
## 
## [[4]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/Sorted_KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7B_rm5_clip.fa.bam"</code></pre>
</div>
<div id="convert-to-beds-find-peaks-and-build-count-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#convert-to-beds-find-peaks-and-build-count-matrix" class="anchor"></a>Convert to beds, find peaks, and build count matrix</h2>
<div id="convert-to-beds" class="section level3">
<h3 class="hasAnchor">
<a href="#convert-to-beds" class="anchor"></a>Convert to beds</h3>
<p><a href="../reference/bamtobed_index.html">CLIPflexR documentation</a></p>
<div class="sourceCode" id="cb42"><pre class="downlit">
<span class="kw">beds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">bams</span>, <span class="kw">bamtobed</span>)
<span class="kw">beds</span>
</pre></div>
<pre><code>## [[1]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/Sorted_KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6A_trim_clip.fa.bed"
## 
## [[2]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/Sorted_KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6B_trim_clip.fa.bed"
## 
## [[3]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/Sorted_KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7A_rm5_clip.fa.bed"
## 
## [[4]]
## [1] "/Users/kathryn/Reprocess_all_paper_datasets/test_processing_vignette/Sorted_KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7B_rm5_clip.fa.bed"</code></pre>
</div>
<div id="make-tag-directories-and-findpeaks-wrapping-homers-maketagdirectory-and-findpeaks" class="section level3">
<h3 class="hasAnchor">
<a href="#make-tag-directories-and-findpeaks-wrapping-homers-maketagdirectory-and-findpeaks" class="anchor"></a>Make tag directories and findpeaks <strong>(wrapping HOMER’s MakeTagDirectory and findpeaks)</strong>
</h3>
<p>Call significant peaks in each sample using R-wrapped <strong>homer_peaks</strong> from <a href="http://homer.ucsd.edu/homer/ngs/tagDir.html">HOMER makeTagDirectory</a> &amp; <a href="http://homer.ucsd.edu/homer/ngs/peaks.html">HOMER findpeaks</a>. <strong>style</strong>: type of HOMER peak calling, “factor” (default); <strong>foldEnrichmentOverLocal</strong>: fold enrichment over local tag count, 2 (default); <strong>localSize</strong>:region to check for local tag enrichment, 10000 (default); <strong>strand</strong>: find peaks using tags on “seperate” (default); <strong>minDist</strong>: minimum distance between peaks, 50 (default); <strong>size</strong>: peak size, 10 (default); <strong>fragLength</strong>: approximate fragment length, 10 (default); <strong>genomeSize</strong>: NULL=2e9 (default, human or mouse), or set for your genome <a href="../reference/homer_peaks.html">CLIPflexR documentation</a></p>
<div class="sourceCode" id="cb44"><pre class="downlit">
<span class="co">#you may want to play with these parameters depending on your protein of interest, and ensure that coverage lines up with the peaks you have called</span>
<span class="kw">peaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">beds</span>, <span class="kw">homer_peaks</span>, fragLength=<span class="fl">25</span>)
</pre></div>
</div>
<div id="make-count-matrix-of-peaks-starting-with-reading-in-homer-peak-files-and-making-a-matrix-that-indicated-which-peaks-were-called-in-each-sample" class="section level3">
<h3 class="hasAnchor">
<a href="#make-count-matrix-of-peaks-starting-with-reading-in-homer-peak-files-and-making-a-matrix-that-indicated-which-peaks-were-called-in-each-sample" class="anchor"></a>Make count matrix of peaks, starting with reading in HOMER peak files and making a matrix that indicated which peaks were called in each sample</h3>
<div class="sourceCode" id="cb45"><pre class="downlit">
<span class="co">#read in peak files and convert to GRanges</span>
<span class="kw">read_peaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">peaks</span>,<span class="fu">function</span>(<span class="kw">x</span>)<span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="kw">x</span>,sep=<span class="st">"\t"</span>,comment.char = <span class="st">"#"</span>,h=<span class="fl">FALSE</span>))
<span class="kw">read_peaksGR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">read_peaks</span>,
                       <span class="fu">function</span>(<span class="kw">x</span>)<span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span>(<span class="kw">x</span><span class="op">$</span><span class="kw">V2</span>,<span class="fu">IRanges</span>(<span class="kw">x</span><span class="op">$</span><span class="kw">V3</span>,<span class="kw">x</span><span class="op">$</span><span class="kw">V4</span>),strand = <span class="kw">x</span><span class="op">$</span><span class="kw">V5</span>))
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">read_peaksGR</span>) <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>( <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="kw">baseDir</span>,<span class="st">"/"</span>), <span class="st">""</span>,  (<span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="kw">peaks</span>)))
<span class="kw">read_peaksGRList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html">GRangesList</a></span>(<span class="kw">read_peaksGR</span>)
<span class="kw">GRSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="kw">read_peaksGRList</span>)
<span class="kw">nGRSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html">reduce</a></span>(<span class="kw">GRSet</span>)
<span class="kw">peakmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span>(<span class="kw">cbind</span>,<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">read_peaksGRList</span>,
                                <span class="fu">function</span>(<span class="kw">x</span>)
                                  <span class="kw">nGRSet</span> <span class="op">%over%</span> <span class="kw">x</span>))
<span class="fu">mcols</span>(<span class="kw">nGRSet</span>) <span class="op">&lt;-</span> <span class="kw">peakmat</span>

<span class="co">#now you have a peakmat where rows are a reduced set of peaks (without an identifier for the time being), and each column  is whether or not the peak was called in that sample</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">peakmat</span>)
</pre></div>
<pre><code>##      Sorted_KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6A_trim_clip.fa/peaks.txt
## [1,]                                                                               FALSE
## [2,]                                                                               FALSE
## [3,]                                                                               FALSE
## [4,]                                                                                TRUE
## [5,]                                                                               FALSE
## [6,]                                                                                TRUE
##      Sorted_KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6B_trim_clip.fa/peaks.txt
## [1,]                                                                               FALSE
## [2,]                                                                               FALSE
## [3,]                                                                               FALSE
## [4,]                                                                               FALSE
## [5,]                                                                                TRUE
## [6,]                                                                               FALSE
##      Sorted_KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7A_rm5_clip.fa/peaks.txt
## [1,]                                                                          FALSE
## [2,]                                                                           TRUE
## [3,]                                                                           TRUE
## [4,]                                                                          FALSE
## [5,]                                                                          FALSE
## [6,]                                                                          FALSE
##      Sorted_KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7B_rm5_clip.fa/peaks.txt
## [1,]                                                                           TRUE
## [2,]                                                                          FALSE
## [3,]                                                                          FALSE
## [4,]                                                                          FALSE
## [5,]                                                                           TRUE
## [6,]                                                                          FALSE</code></pre>
</div>
<div id="the-next-step-is-to-countfrombed-to-append-the-number-of-reads-at-each-peak-location-by-sample" class="section level3">
<h3 class="hasAnchor">
<a href="#the-next-step-is-to-countfrombed-to-append-the-number-of-reads-at-each-peak-location-by-sample" class="anchor"></a>The next step is to countFromBed to append the number of reads at each peak location, by sample</h3>
<p><a href="../reference/countFromBed.html">CLIPflexR documentation</a></p>
<div class="sourceCode" id="cb47"><pre class="downlit">
<span class="co">#defaults are: notStranded=TRUE,interFeature=FALSE</span>
<span class="kw">countmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">beds</span>,<span class="kw">countFromBed</span>,GR=<span class="kw">nGRSet</span>)
</pre></div>
<pre><code>## Warning in .Seqinfo.mergexy(x, y): Each of the 2 combined objects has sequence levels not in the other:
##   - in 'x': NIGP01000917, NIGP01002025
##   - in 'y': NIGP01002089, NIGP01000727, NIGP01000475, NIGP01002221, NIGP01001492, NIGP01000892, NIGP01000201, NIGP01001727, NIGP01001982, NIGP01000101, NIGP01002078, NIGP01001086, NIGP01002082, NIGP01002045, NIGP01002203, NIGP01000451, NIGP01000495, NIGP01002020, NIGP01001844, NIGP01002286, NIGP01002127, NIGP01000184, NIGP01000944, NIGP01000304, NIGP01000130, NIGP01001935, NIGP01000647, NIGP01002035, NIGP01001914, NIGP01001774, NIGP01000058, NIGP01000662, NIGP01000976, NIGP01000253, NIGP01001770, NIGP01000416, NIGP01001671, NIGP01001712, NIGP01002113, NIGP01001514, NIGP01000379, NIGP01000036, NIGP01002034, NIGP01000552, NIGP01000094, NIGP01001837, NIGP01001551, NIGP01000525, NIGP01002215, NIGP01001678, NIGP01000638, NIGP01001599, NIGP01000106, NIGP01001294, NIGP01002178, NIGP01000256, NIGP01000283, NIGP01001031, NIGP01002155, NIGP01000077, NIGP01001793, NIGP01000371, NIGP01000658, NIGP01001480, NIGP01000848, NIGP01000415, NIGP01000180, NIGP01001039, NIGP01000110, NIGP01000242, NIGP01001503, NIGP01001976, NIGP01000537, NIGP01000565, NIGP01000807, NIGP01002168, NIGP01000482, NIGP01002289, NIGP01000751, NIGP01001945, NIGP01000574, NIGP01002209, NIGP01001759, NIGP01000402, NIGP01000198, NIGP01000292, NIGP01002029, NIGP01000294, NIGP01000909, NIGP01000139, NIGP01000282, NIGP01001786, NIGP01000376, NIGP01001698, NIGP01001206, NIGP01002240, NIGP01001030, NIGP01001414, NIGP01001750, NIGP01002188, NIGP01000072, NIGP01000080, NIGP01002269, NIGP01000033, NIGP01001661, NIGP01002234, NIGP01000761, NIGP01000063, NIGP01002251, NIGP01002176, NIGP01000433, NIGP01000034, NIGP01000159, NIGP01000575, NIGP01002300, NIGP01001458, NIGP01001507, NIGP01000182, NIGP01000941, NIGP01000188, NIGP01002297, NIGP01000697, NIGP01000629, NIGP01001062, NIGP01001508, NIGP01002230, NIGP01000197, NIGP01000266, NIGP01000699, NIGP01000541, NIGP01001755, NIGP01001696, NIGP01000656, NIGP01000563, NIGP01000888, NIGP01000327, NIGP01001988, NIGP01001641, NIGP01001846, NIGP01001908, NIGP01000986, NIGP01001769, NIGP01000440, NIGP01002046, NIGP01000505, NIGP01002152, NIGP01000627, NIGP01000598, NIGP01000274, NIGP01000414, NIGP01002253, NIGP01001029, NIGP01000599, NIGP01001510, NIGP01001887, NIGP01001615, NIGP01000650, NIGP01000735, NIGP01002199, NIGP01001036, NIGP01002153, NIGP01001949, NIGP01002196, NIGP01000407, NIGP01000437, NIGP01001919, NIGP01000809, NIGP01001929, NIGP01001703, NIGP01001866, NIGP01000504, NIGP01001327, NIGP01002058, NIGP01000779, NIGP01001557, NIGP01002024, NIGP01000859, NIGP01000206, NIGP01000189, NIGP01000210, NIGP01001019, NIGP01001717, NIGP01002087, NIGP01001650, NIGP01001845, NIGP01000361, NIGP01000579, NIGP01001741, NIGP01000205, NIGP01000802, NIGP01000548, NIGP01001757, NIGP01000356, NIGP01000893, NIGP01000637, NIGP01001469, NIGP01000780, NIGP01001915, NIGP01001943, NIGP01000335, NIGP01001933, NIGP01000920, NIGP01000863, NIGP01000417, NIGP01000845, NIGP01001008, NIGP01001803, NIGP01000039, NIGP01001816, NIGP01000665, NIGP01000818, NIGP01000521, NIGP01000578, NIGP01000532, NIGP01000821, NIGP01001546, NIGP01002242, NIGP01000384, NIGP01000295, NIGP01000408, NIGP01001940, NIGP01001566, NIGP01000865, NIGP01000646, NIGP01000187, NIGP01000962, NIGP01001024, NIGP01001076, NIGP01000497, NIGP01000916, NIGP01001927, NIGP01000791, NIGP01000760, NIGP01001028, NIGP01000124, NIGP01002055, NIGP01000212, NIGP01001898, NIGP01000149, NIGP01002044, NIGP01000363, NIGP01000904, NIGP01002133, NIGP01000123, NIGP01001567, NIGP01001956, NIGP01000232, NIGP01002275, NIGP01000142, NIGP01000413, NIGP01001736, NIGP01000728, NIGP01000608, NIGP01000290, NIGP01000477, NIGP01000806, NIGP01000315, NIGP01001858, NIGP01000577, NIGP01001066, NIGP01001934, NIGP01001767, NIGP01000573, NIGP01000648, NIGP01000747, NIGP01000397, NIGP01002103, NIGP01000600, NIGP01002121, NIGP01000254, NIGP01001977, NIGP01000959, NIGP01000009, NIGP01000640, NIGP01000061, NIGP01002278, NIGP01001893, NIGP01001584, NIGP01000218, NIGP01000148, NIGP01000711, NIGP01001875, NIGP01001085, NIGP01000463, NIGP01001536, NIGP01001672, NIGP01000450, NIGP01000736, NIGP01000281, NIGP01000031, NIGP01001758, NIGP01000840, NIGP01001961, NIGP01001925, NIGP01001554, NIGP01001764, NIGP01000098, NIGP01000457, NIGP01001690, NIGP01001534, NIGP01001942, NIGP01000216, NIGP01000965, NIGP01000328, NIGP01001909, NIGP01001392, NIGP01000522, NIGP01000213, NIGP01001593, NIGP01000464, NIGP01000456, NIGP01001540, NIGP01002260, NIGP01000279, NIGP01001061, NIGP01002191, NIGP01002064, NIGP01000486, NIGP01000775, NIGP01001847, NIGP01001688, NIGP01000963, NIGP01002181, NIGP01002014, NIGP01000424, NIGP01002179, NIGP01001077, NIGP01000870, NIGP01000580, NIGP01000056, NIGP01001498, NIGP01000639, NIGP01000768, NIGP01000745, NIGP01001781, NIGP01000813, NIGP01000666, NIGP01002099, NIGP01000092, NIGP01000885, NIGP01000700, NIGP01001634, NIGP01000381, NIGP01000645, NIGP01000958, NIGP01000620, NIGP01000263, NIGP01000896, NIGP01000046, NIGP01000025, NIGP01000378, NIGP01000403, NIGP01001639, NIGP01000185, NIGP01001561, NIGP01001601, NIGP01002009, NIGP01000095, NIGP01001704, NIGP01001053, NIGP01000194, NIGP01000866, NIGP01001497, NIGP01000568, NIGP01001017, NIGP01002126, NIGP01000369, NIGP01000020, NIGP01001573, NIGP01000291, NIGP01002170, NIGP01001430, NIGP01002096, NIGP01002167, NIGP01000257, NIGP01000996, NIGP01000670, NIGP01001810, NIGP01000476, NIGP01001928, NIGP01001082, NIGP01000950, NIGP01000759, NIGP01001635, NIGP01000737, NIGP01001415, NIGP01000060, NIGP01002207, NIGP01002077, NIGP01000680, NIGP01000052, NIGP01000121, NIGP01000113, NIGP01000144, NIGP01002154, NIGP01000975, NIGP01002157, NIGP01000269, NIGP01001072, NIGP01000088, NIGP01000583, NIGP01000426, NIGP01002163, NIGP01000581, NIGP01000125, NIGP01000247, NIGP01001791, NIGP01001499, NIGP01001676, NIGP01001711, NIGP01001902, NIGP01000480, NIGP01001035, NIGP01000709, NIGP01001763, NIGP01000367, NIGP01000597, NIGP01001068, NIGP01002146, NIGP01000473, NIGP01001748, NIGP01001728, NIGP01000823, NIGP01000103, NIGP01002106, NIGP01000014, NIGP01000134, NIGP01002065, NIGP01001707, NIGP01000203, NIGP01000657, NIGP01002263, NIGP01001737, NIGP01002169, NIGP01001819, NIGP01000280, NIGP01002151, NIGP01001833, NIGP01001708, NIGP01002273, NIGP01001084, NIGP01001059, NIGP01000596, NIGP01001807, NIGP01000375, NIGP01000507, NIGP01000714, NIGP01001673, NIGP01001880, NIGP01000852, NIGP01002042, NIGP01000489, NIGP01000757, NIGP01000340, NIGP01000447, NIGP01001805, NIGP01000374, NIGP01001700, NIGP01000713, NIGP01000069, NIGP01002122, NIGP01000734, NIGP01000047, NIGP01001958, NIGP01001890, NIGP01000453, NIGP01001994, NIGP01000805, NIGP01001749, NIGP01001979, NIGP01000621, NIGP01000694, NIGP01000972, NIGP01001968, NIGP01001839, NIGP01001989, NIGP01001960, NIGP01000209, NIGP01002187, NIGP01002001, NIGP01001813, NIGP01000150, NIGP01000991, NIGP01002084, NIGP01000514, NIGP01002039, NIGP01001739, NIGP01000157, NIGP01001842, NIGP01001073, NIGP01000005, NIGP01000285, NIGP01001281, NIGP01001572, NIGP01000786, NIGP01001867, NIGP01002158, NIGP01001521, NIGP01000084, NIGP01001517, NIGP01002189, NIGP01000919, NIGP01001530, NIGP01000635, NIGP01000107, NIGP01000204, NIGP01000644, NIGP01001014, NIGP01002228, NIGP01000228, NIGP01002264, NIGP01000557, NIGP01000329, NIGP01001630, NIGP01002110, NIGP01000246, NIGP01000049, NIGP01002128, NIGP01001439, NIGP01002101, NIGP01001952, NIGP01000999, NIGP01000411, NIGP01001146, NIGP01001074, NIGP01000750, NIGP01000096, NIGP01001811, NIGP01002308, NIGP01000718, NIGP01002193, NIGP01000938, NIGP01000053, NIGP01001041, NIGP01001595, NIGP01001752, NIGP01001733, NIGP01000547, NIGP01000710, NIGP01002177, NIGP01001230, NIGP01001533, NIGP01001691, NIGP01001301, NIGP01000667, NIGP01000617, NIGP01000441, NIGP01001240, NIGP01001991, NIGP01001046, NIGP01000041, NIGP01002130, NIGP01000633, NIGP01000907, NIGP01001018, NIGP01000832, NIGP01001519, NIGP01001153, NIGP01000017, NIGP01001217, NIGP01000654, NIGP01001795, NIGP01000422, NIGP01000967, NIGP01000942, NIGP01000609, NIGP01000560, NIGP01000401, NIGP01001081, NIGP01000652, NIGP01001783, NIGP01000177, NIGP01000882, NIGP01001067, NIGP01000326, NIGP01002229, NIGP01</code></pre>
<pre><code>## Warning in .Seqinfo.mergexy(x, y): Each of the 2 combined objects has sequence levels not in the other:
##   - in 'x': NIGP01001735
##   - in 'y': NIGP01002089, NIGP01000727, NIGP01000475, NIGP01002221, NIGP01001492, NIGP01000892, NIGP01000201, NIGP01001727, NIGP01001982, NIGP01000101, NIGP01002078, NIGP01001086, NIGP01002082, NIGP01002045, NIGP01002203, NIGP01000451, NIGP01000495, NIGP01002020, NIGP01001844, NIGP01002286, NIGP01002127, NIGP01000184, NIGP01000944, NIGP01000304, NIGP01000130, NIGP01001935, NIGP01000647, NIGP01002035, NIGP01001914, NIGP01001774, NIGP01000058, NIGP01000662, NIGP01000976, NIGP01000253, NIGP01001770, NIGP01000416, NIGP01001671, NIGP01001712, NIGP01002113, NIGP01001514, NIGP01000379, NIGP01000036, NIGP01002034, NIGP01000552, NIGP01000094, NIGP01001837, NIGP01001551, NIGP01000525, NIGP01002215, NIGP01001678, NIGP01000638, NIGP01001599, NIGP01000106, NIGP01001294, NIGP01002178, NIGP01000256, NIGP01000283, NIGP01001031, NIGP01002155, NIGP01000077, NIGP01001793, NIGP01000371, NIGP01000658, NIGP01001480, NIGP01000848, NIGP01000415, NIGP01001039, NIGP01000110, NIGP01000242, NIGP01001503, NIGP01001976, NIGP01000537, NIGP01000565, NIGP01000807, NIGP01002168, NIGP01000593, NIGP01000482, NIGP01002289, NIGP01000751, NIGP01001945, NIGP01000574, NIGP01002209, NIGP01001759, NIGP01000402, NIGP01000198, NIGP01000432, NIGP01000292, NIGP01002029, NIGP01000294, NIGP01000909, NIGP01000139, NIGP01000282, NIGP01001786, NIGP01000376, NIGP01001698, NIGP01001206, NIGP01002240, NIGP01001030, NIGP01001414, NIGP01001750, NIGP01002188, NIGP01000072, NIGP01000080, NIGP01002269, NIGP01000033, NIGP01001661, NIGP01002234, NIGP01000602, NIGP01000761, NIGP01000063, NIGP01002251, NIGP01002176, NIGP01000433, NIGP01000034, NIGP01000159, NIGP01000575, NIGP01002300, NIGP01001458, NIGP01001507, NIGP01000941, NIGP01000188, NIGP01002297, NIGP01000629, NIGP01001062, NIGP01001508, NIGP01002230, NIGP01000197, NIGP01000699, NIGP01000541, NIGP01001755, NIGP01000656, NIGP01000563, NIGP01000888, NIGP01000327, NIGP01001988, NIGP01001641, NIGP01001846, NIGP01001908, NIGP01000986, NIGP01001769, NIGP01000440, NIGP01002046, NIGP01002152, NIGP01000627, NIGP01000274, NIGP01000414, NIGP01002253, NIGP01001029, NIGP01000599, NIGP01001510, NIGP01001887, NIGP01001615, NIGP01000650, NIGP01000735, NIGP01002199, NIGP01001036, NIGP01002153, NIGP01001949, NIGP01002196, NIGP01000407, NIGP01000437, NIGP01001919, NIGP01000809, NIGP01001929, NIGP01001703, NIGP01001866, NIGP01000504, NIGP01001327, NIGP01002058, NIGP01000779, NIGP01001557, NIGP01000859, NIGP01000206, NIGP01000189, NIGP01000210, NIGP01001019, NIGP01001717, NIGP01002087, NIGP01001650, NIGP01001845, NIGP01000361, NIGP01000579, NIGP01001741, NIGP01000205, NIGP01000802, NIGP01000548, NIGP01000356, NIGP01000893, NIGP01001469, NIGP01000780, NIGP01001915, NIGP01001943, NIGP01000335, NIGP01001933, NIGP01000920, NIGP01000863, NIGP01000417, NIGP01000845, NIGP01001008, NIGP01001803, NIGP01000039, NIGP01001816, NIGP01000665, NIGP01000818, NIGP01000521, NIGP01000578, NIGP01000532, NIGP01000821, NIGP01001546, NIGP01002242, NIGP01000384, NIGP01000295, NIGP01000471, NIGP01000408, NIGP01001940, NIGP01001566, NIGP01000865, NIGP01000646, NIGP01000187, NIGP01000316, NIGP01001024, NIGP01001076, NIGP01000497, NIGP01000916, NIGP01000791, NIGP01000760, NIGP01001028, NIGP01000124, NIGP01001612, NIGP01002055, NIGP01000212, NIGP01001898, NIGP01000149, NIGP01002044, NIGP01000904, NIGP01000123, NIGP01001697, NIGP01001567, NIGP01001956, NIGP01000232, NIGP01002275, NIGP01000142, NIGP01000413, NIGP01001736, NIGP01000728, NIGP01000608, NIGP01000290, NIGP01000477, NIGP01000806, NIGP01000315, NIGP01001858, NIGP01000577, NIGP01001066, NIGP01001934, NIGP01001767, NIGP01000573, NIGP01000648, NIGP01000747, NIGP01000397, NIGP01002103, NIGP01000600, NIGP01002121, NIGP01000254, NIGP01001977, NIGP01000959, NIGP01000009, NIGP01000640, NIGP01001060, NIGP01002278, NIGP01001893, NIGP01000218, NIGP01000148, NIGP01000711, NIGP01001875, NIGP01001085, NIGP01001536, NIGP01001672, NIGP01000736, NIGP01000281, NIGP01000031, NIGP01001758, NIGP01000840, NIGP01001516, NIGP01001961, NIGP01001925, NIGP01001554, NIGP01001606, NIGP01000673, NIGP01001764, NIGP01002054, NIGP01000098, NIGP01000457, NIGP01002095, NIGP01001690, NIGP01001534, NIGP01001942, NIGP01000216, NIGP01000328, NIGP01001909, NIGP01001392, NIGP01001058, NIGP01000512, NIGP01000522, NIGP01000213, NIGP01000464, NIGP01000456, NIGP01001540, NIGP01002260, NIGP01001061, NIGP01002191, NIGP01002064, NIGP01000486, NIGP01000775, NIGP01001847, NIGP01001688, NIGP01001950, NIGP01000963, NIGP01002181, NIGP01002014, NIGP01000424, NIGP01002179, NIGP01001077, NIGP01000870, NIGP01000056, NIGP01001498, NIGP01000639, NIGP01000768, NIGP01000745, NIGP01001781, NIGP01000813, NIGP01000666, NIGP01000087, NIGP01002099, NIGP01000092, NIGP01000885, NIGP01000700, NIGP01001634, NIGP01000381, NIGP01000645, NIGP01000958, NIGP01000620, NIGP01000263, NIGP01000896, NIGP01000025, NIGP01000378, NIGP01000403, NIGP01001639, NIGP01000185, NIGP01001561, NIGP01001601, NIGP01002009, NIGP01000095, NIGP01001704, NIGP01000217, NIGP01001053, NIGP01000194, NIGP01000866, NIGP01001497, NIGP01002126, NIGP01000020, NIGP01001573, NIGP01000291, NIGP01002170, NIGP01001114, NIGP01000289, NIGP01001683, NIGP01002096, NIGP01002167, NIGP01000257, NIGP01000670, NIGP01001810, NIGP01000636, NIGP01000476, NIGP01001928, NIGP01001082, NIGP01000950, NIGP01000759, NIGP01001635, NIGP01000737, NIGP01001415, NIGP01000060, NIGP01002207, NIGP01002077, NIGP01000386, NIGP01000052, NIGP01000121, NIGP01000113, NIGP01000144, NIGP01002154, NIGP01000190, NIGP01002157, NIGP01000269, NIGP01001072, NIGP01000088, NIGP01000583, NIGP01000426, NIGP01002163, NIGP01000581, NIGP01000125, NIGP01000247, NIGP01001791, NIGP01001499, NIGP01001676, NIGP01001711, NIGP01001902, NIGP01000523, NIGP01001870, NIGP01001035, NIGP01000709, NIGP01001763, NIGP01000367, NIGP01000597, NIGP01001068, NIGP01002146, NIGP01000473, NIGP01001748, NIGP01001728, NIGP01000823, NIGP01000103, NIGP01002106, NIGP01000014, NIGP01000134, NIGP01002065, NIGP01000203, NIGP01002263, NIGP01001737, NIGP01001819, NIGP01002151, NIGP01001833, NIGP01000969, NIGP01002273, NIGP01001059, NIGP01002283, NIGP01001807, NIGP01000375, NIGP01000507, NIGP01000714, NIGP01001880, NIGP01000852, NIGP01000348, NIGP01002042, NIGP01000344, NIGP01000489, NIGP01000757, NIGP01000340, NIGP01000447, NIGP01001805, NIGP01000374, NIGP01000713, NIGP01000069, NIGP01002122, NIGP01000734, NIGP01000047, NIGP01001958, NIGP01001890, NIGP01000453, NIGP01001994, NIGP01000805, NIGP01001749, NIGP01002180, NIGP01001979, NIGP01000621, NIGP01000694, NIGP01000972, NIGP01001968, NIGP01001839, NIGP01000449, NIGP01001989, NIGP01001960, NIGP01000209, NIGP01000156, NIGP01002001, NIGP01000150, NIGP01000991, NIGP01000514, NIGP01000871, NIGP01001440, NIGP01000157, NIGP01001842, NIGP01000285, NIGP01001281, NIGP01001572, NIGP01000786, NIGP01001867, NIGP01002158, NIGP01001521, NIGP01000084, NIGP01002086, NIGP01001517, NIGP01002189, NIGP01000919, NIGP01001530, NIGP01000635, NIGP01000107, NIGP01000204, NIGP01001785, NIGP01000644, NIGP01001014, NIGP01000228, NIGP01002264, NIGP01000557, NIGP01000329, NIGP01000539, NIGP01001630, NIGP01001049, NIGP01002110, NIGP01002128, NIGP01001439, NIGP01002101, NIGP01002129, NIGP01001952, NIGP01000999, NIGP01000411, NIGP01000750, NIGP01000096, NIGP01001811, NIGP01002308, NIGP01000718, NIGP01002193, NIGP01001047, NIGP01000938, NIGP01000053, NIGP01001041, NIGP01001595, NIGP01001752, NIGP01000547, NIGP01000710, NIGP01002288, NIGP01002177, NIGP01001533, NIGP01001838, NIGP01002186, NIGP01001691, NIGP01001301, NIGP01000667, NIGP01000617, NIGP01000441, NIGP01001240, NIGP01001991, NIGP01001046, NIGP01000041, NIGP01002130, NIGP01000633, NIGP01000907, NIGP01001018, NIGP01001519, NIGP01000017, NIGP01001217, NIGP01000654, NIGP01001795, NIGP01000422, NIGP01000967, NIGP01000942, NIGP01000609, NIGP01000428, NIGP01000560, NIGP01000401, NIGP01001081, NIGP01000177, NIGP01000882, NIGP01001067, NIGP01000326, NIGP01000843, NIGP01002118, NIGP01000663, NIGP01000310, NIGP01000233, NIGP01001828, NIGP01000465, NIGP01000953, NIGP01000649, NIGP01002298, NIGP01000394, NIGP01001743, NIGP01001558, NIGP01000377, NIGP01000970, NIGP01</code></pre>
<pre><code>## Warning in .Seqinfo.mergexy(x, y): Each of the 2 combined objects has sequence levels not in the other:
##   - in 'x': NIGP01000990
##   - in 'y': NIGP01002089, NIGP01000727, NIGP01000475, NIGP01002221, NIGP01001492, NIGP01000892, NIGP01000201, NIGP01001727, NIGP01001982, NIGP01000101, NIGP01002078, NIGP01001086, NIGP01002082, NIGP01002045, NIGP01002203, NIGP01000451, NIGP01000495, NIGP01002020, NIGP01001844, NIGP01002286, NIGP01002127, NIGP01000184, NIGP01000944, NIGP01000304, NIGP01000130, NIGP01001935, NIGP01000647, NIGP01002035, NIGP01001914, NIGP01001774, NIGP01000058, NIGP01000662, NIGP01000976, NIGP01000253, NIGP01001770, NIGP01000416, NIGP01001671, NIGP01001712, NIGP01002113, NIGP01001514, NIGP01000379, NIGP01000036, NIGP01002034, NIGP01000552, NIGP01001837, NIGP01001551, NIGP01000525, NIGP01002215, NIGP01001678, NIGP01000638, NIGP01001599, NIGP01000106, NIGP01001294, NIGP01002178, NIGP01000256, NIGP01000283, NIGP01001031, NIGP01002155, NIGP01000077, NIGP01001793, NIGP01000371, NIGP01000658, NIGP01001480, NIGP01000848, NIGP01000415, NIGP01000180, NIGP01001039, NIGP01000110, NIGP01000242, NIGP01001503, NIGP01001976, NIGP01000537, NIGP01000565, NIGP01000807, NIGP01002168, NIGP01000593, NIGP01000482, NIGP01002289, NIGP01000751, NIGP01001945, NIGP01000574, NIGP01002209, NIGP01001759, NIGP01000402, NIGP01000198, NIGP01000432, NIGP01002029, NIGP01000294, NIGP01000909, NIGP01000139, NIGP01000282, NIGP01001786, NIGP01000376, NIGP01001698, NIGP01001206, NIGP01002240, NIGP01001030, NIGP01001414, NIGP01001750, NIGP01002188, NIGP01000072, NIGP01000080, NIGP01002269, NIGP01000033, NIGP01001661, NIGP01002234, NIGP01000602, NIGP01000761, NIGP01000063, NIGP01002251, NIGP01002176, NIGP01000433, NIGP01000034, NIGP01000159, NIGP01002300, NIGP01001458, NIGP01001507, NIGP01000182, NIGP01000941, NIGP01000188, NIGP01002297, NIGP01000697, NIGP01000629, NIGP01001062, NIGP01001508, NIGP01002230, NIGP01000197, NIGP01000266, NIGP01000699, NIGP01000541, NIGP01001755, NIGP01001696, NIGP01000656, NIGP01000563, NIGP01000888, NIGP01000327, NIGP01001988, NIGP01001641, NIGP01001846, NIGP01001908, NIGP01000986, NIGP01001769, NIGP01000440, NIGP01002046, NIGP01000505, NIGP01002152, NIGP01000627, NIGP01000598, NIGP01000274, NIGP01000414, NIGP01002253, NIGP01001029, NIGP01000599, NIGP01001510, NIGP01001887, NIGP01000650, NIGP01000735, NIGP01002199, NIGP01001036, NIGP01002153, NIGP01001949, NIGP01002196, NIGP01000407, NIGP01001919, NIGP01000809, NIGP01001929, NIGP01001703, NIGP01001866, NIGP01002058, NIGP01000779, NIGP01001557, NIGP01002024, NIGP01000859, NIGP01000206, NIGP01000189, NIGP01000210, NIGP01001019, NIGP01001717, NIGP01002087, NIGP01001650, NIGP01001965, NIGP01001845, NIGP01000361, NIGP01000579, NIGP01001741, NIGP01000205, NIGP01000802, NIGP01001757, NIGP01000356, NIGP01000893, NIGP01000637, NIGP01001469, NIGP01000780, NIGP01001915, NIGP01001943, NIGP01000335, NIGP01001933, NIGP01000920, NIGP01000863, NIGP01000417, NIGP01000845, NIGP01001008, NIGP01000039, NIGP01001816, NIGP01000665, NIGP01000818, NIGP01000521, NIGP01000578, NIGP01000532, NIGP01000821, NIGP01001546, NIGP01002242, NIGP01000384, NIGP01000295, NIGP01000408, NIGP01001940, NIGP01001566, NIGP01000865, NIGP01000187, NIGP01000962, NIGP01000316, NIGP01001024, NIGP01001076, NIGP01000497, NIGP01000916, NIGP01001927, NIGP01000791, NIGP01000124, NIGP01001612, NIGP01002055, NIGP01000212, NIGP01001898, NIGP01002044, NIGP01000363, NIGP01000904, NIGP01002133, NIGP01000123, NIGP01001697, NIGP01001567, NIGP01001956, NIGP01002275, NIGP01000142, NIGP01000413, NIGP01001736, NIGP01000728, NIGP01000290, NIGP01000477, NIGP01000806, NIGP01000315, NIGP01001858, NIGP01000577, NIGP01001066, NIGP01001934, NIGP01001767, NIGP01000648, NIGP01000747, NIGP01000397, NIGP01002103, NIGP01000600, NIGP01002121, NIGP01000254, NIGP01001977, NIGP01000959, NIGP01000009, NIGP01000640, NIGP01000061, NIGP01001893, NIGP01000218, NIGP01000711, NIGP01001875, NIGP01001085, NIGP01000463, NIGP01001672, NIGP01000450, NIGP01000281, NIGP01000031, NIGP01001758, NIGP01000840, NIGP01001961, NIGP01001925, NIGP01001554, NIGP01001606, NIGP01001764, NIGP01002054, NIGP01000098, NIGP01000457, NIGP01002095, NIGP01001690, NIGP01001534, NIGP01001942, NIGP01000216, NIGP01000965, NIGP01000328, NIGP01001909, NIGP01001392, NIGP01001058, NIGP01000512, NIGP01000213, NIGP01001593, NIGP01000464, NIGP01000456, NIGP01001540, NIGP01002260, NIGP01000279, NIGP01001061, NIGP01002064, NIGP01000775, NIGP01001847, NIGP01001950, NIGP01000963, NIGP01002181, NIGP01000424, NIGP01002179, NIGP01001077, NIGP01000870, NIGP01000580, NIGP01000056, NIGP01001498, NIGP01000639, NIGP01000768, NIGP01000745, NIGP01001781, NIGP01000666, NIGP01000087, NIGP01002099, NIGP01000092, NIGP01000885, NIGP01000700, NIGP01001634, NIGP01000381, NIGP01000645, NIGP01000958, NIGP01000263, NIGP01000896, NIGP01000046, NIGP01000378, NIGP01000403, NIGP01001639, NIGP01001561, NIGP01002009, NIGP01000095, NIGP01001704, NIGP01000217, NIGP01001053, NIGP01000194, NIGP01000866, NIGP01001497, NIGP01002126, NIGP01001573, NIGP01001114, NIGP01000289, NIGP01002096, NIGP01002167, NIGP01000257, NIGP01000996, NIGP01000670, NIGP01001810, NIGP01000476, NIGP01001928, NIGP01001082, NIGP01000950, NIGP01000759, NIGP01001635, NIGP01000737, NIGP01001415, NIGP01000060, NIGP01002207, NIGP01002077, NIGP01000386, NIGP01000052, NIGP01000121, NIGP01000113, NIGP01000144, NIGP01002157, NIGP01000269, NIGP01000088, NIGP01000583, NIGP01000426, NIGP01000167, NIGP01002163, NIGP01000581, NIGP01000125, NIGP01000247, NIGP01001791, NIGP01001499, NIGP01001676, NIGP01001711, NIGP01001902, NIGP01000523, NIGP01000480, NIGP01001870, NIGP01000709, NIGP01000367, NIGP01001068, NIGP01002146, NIGP01000473, NIGP01001748, NIGP01001728, NIGP01000823, NIGP01002106, NIGP01000014, NIGP01000134, NIGP01002065, NIGP01000203, NIGP01000657, NIGP01002263, NIGP01001737, NIGP01001819, NIGP01002151, NIGP01001833, NIGP01000969, NIGP01001059, NIGP01000596, NIGP01001807, NIGP01000375, NIGP01000507, NIGP01000714, NIGP01001880, NIGP01002042, NIGP01000344, NIGP01000489, NIGP01000757, NIGP01000447, NIGP01001805, NIGP01000374, NIGP01000713, NIGP01002122, NIGP01000734, NIGP01000047, NIGP01001958, NIGP01000453, NIGP01000805, NIGP01001749, NIGP01002180, NIGP01001979, NIGP01000621, NIGP01000694, NIGP01000972, NIGP01001968, NIGP01001839, NIGP01001989, NIGP01001960, NIGP01000209, NIGP01002187, NIGP01000156, NIGP01002001, NIGP01001813, NIGP01000150, NIGP01000991, NIGP01002084, NIGP01000514, NIGP01002039, NIGP01000871, NIGP01001440, NIGP01001739, NIGP01001713, NIGP01000157, NIGP01001842, NIGP01001073, NIGP01000005, NIGP01000285, NIGP01001572, NIGP01000786, NIGP01001867, NIGP01001521, NIGP01000084, NIGP01002086, NIGP01001338, NIGP01001517, NIGP01000919, NIGP01000635, NIGP01000107, NIGP01001785, NIGP01001014, NIGP01002228, NIGP01000228, NIGP01002264, NIGP01000329, NIGP01000539, NIGP01001630, NIGP01001049, NIGP01002110, NIGP01002128, NIGP01001439, NIGP01002101, NIGP01002129, NIGP01001952, NIGP01000411, NIGP01001146, NIGP01001074, NIGP01000750, NIGP01000096, NIGP01001811, NIGP01000718, NIGP01002193, NIGP01000938, NIGP01000053, NIGP01001041, NIGP01001595, NIGP01001752, NIGP01000547, NIGP01000710, NIGP01002288, NIGP01002177, NIGP01001533, NIGP01001838, NIGP01001691, NIGP01001301, NIGP01000667, NIGP01000617, NIGP01000441, NIGP01001240, NIGP01001991, NIGP01001046, NIGP01000041, NIGP01002130, NIGP01000633, NIGP01000907, NIGP01001018, NIGP01001519, NIGP01001821, NIGP01001153, NIGP01001217, NIGP01000654, NIGP01001795, NIGP01000422, NIGP01000967, NIGP01000942, NIGP01000609, NIGP01000560, NIGP01000401, NIGP01001081, NIGP01000652, NIGP01000055, NIGP01000177, NIGP01000882, NIGP01001067, NIGP01000326, NIGP01002229, NIGP01000616, NIGP01000238, NIGP01000742, NIGP01000843, NIGP01002118, NIGP01001178, NIGP01000310, NIGP01000979, NIGP01000233, NIGP01000465, NIGP01000953, NIGP01000649, NIGP01001993, NIGP01002298, NIGP01001743, NIGP01001558, NIGP01000377, NIGP01000970, NIGP01000948, NIGP01001216, NIGP01000436, NIGP01000715, NIGP01000261, NIGP01001901, NIGP01000481, NIGP01000674, NIGP01001777, NIGP01000877, NIGP01000632, NIGP01000546, NIGP01000752, NIGP01000010, NIGP01000421, NIGP01000501, NIGP01000082, NIGP01000089, NIGP01001897, NIGP01001710, NIGP01</code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit">
<span class="kw">countmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span>(<span class="kw">cbind</span>,<span class="kw">countmat</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="kw">countmat</span>) <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="kw">beds</span>))

<span class="co">#now bind nGRSet, a GRanges with peak ranges and TRUE/FALSE at each peak, with countmat, the number of reads at each peak</span>
<span class="fu">mcols</span>(<span class="kw">nGRSet</span>) <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="fu">mcols</span>(<span class="kw">nGRSet</span>)),<span class="kw">countmat</span>)
<span class="kw">peaksDF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="kw">nGRSet</span>)
<span class="kw">peaksDF</span><span class="op">$</span><span class="kw">peakID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="kw">peaksDF</span><span class="op">$</span><span class="kw">seqnames</span>, <span class="st">":"</span>, <span class="kw">peaksDF</span><span class="op">$</span><span class="kw">start</span>, <span class="st">"_"</span>, <span class="kw">peaksDF</span><span class="op">$</span><span class="kw">end</span>, <span class="st">":"</span>, <span class="kw">peaksDF</span><span class="op">$</span><span class="kw">strand</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">peaksDF</span>)
</pre></div>
<pre><code>##   seqnames   start     end width strand
## 1        1  301043  301053    11      +
## 2        1  580903  580913    11      +
## 3        1 1183109 1183119    11      +
## 4        1 1386212 1386222    11      +
## 5        1 1947557 1947569    13      +
## 6        1 1947625 1947635    11      +
##   Sorted_KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6A_trim_clip.fa.peaks.txt
## 1                                                                               FALSE
## 2                                                                               FALSE
## 3                                                                               FALSE
## 4                                                                                TRUE
## 5                                                                               FALSE
## 6                                                                                TRUE
##   Sorted_KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6B_trim_clip.fa.peaks.txt
## 1                                                                               FALSE
## 2                                                                               FALSE
## 3                                                                               FALSE
## 4                                                                               FALSE
## 5                                                                                TRUE
## 6                                                                               FALSE
##   Sorted_KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7A_rm5_clip.fa.peaks.txt
## 1                                                                          FALSE
## 2                                                                           TRUE
## 3                                                                           TRUE
## 4                                                                          FALSE
## 5                                                                          FALSE
## 6                                                                          FALSE
##   Sorted_KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7B_rm5_clip.fa.peaks.txt
## 1                                                                           TRUE
## 2                                                                          FALSE
## 3                                                                          FALSE
## 4                                                                          FALSE
## 5                                                                           TRUE
## 6                                                                          FALSE
##   Sorted_KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6A_trim_clip.fa.bed
## 1                                                                             0
## 2                                                                             0
## 3                                                                             0
## 4                                                                             3
## 5                                                                             3
## 6                                                                             3
##   Sorted_KRG091819_S1_L001_R1_001_collapse_rm5_Aag2_Ago1_Exp6B_trim_clip.fa.bed
## 1                                                                             0
## 2                                                                             0
## 3                                                                             0
## 4                                                                             1
## 5                                                                             4
## 6                                                                             1
##   Sorted_KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7A_rm5_clip.fa.bed
## 1                                                                        0
## 2                                                                        3
## 3                                                                        3
## 4                                                                        0
## 5                                                                        5
## 6                                                                        2
##   Sorted_KRG092419_S1_L001_R1_001_collapse_Aag2_Ago1_Exp7B_rm5_clip.fa.bed
## 1                                                                        3
## 2                                                                        0
## 3                                                                        3
## 4                                                                        0
## 5                                                                        5
## 6                                                                        0
##                peakID
## 1   1:301043_301053:+
## 2   1:580903_580913:+
## 3 1:1183109_1183119:+
## 4 1:1386212_1386222:+
## 5 1:1947557_1947569:+
## 6 1:1947625_1947635:+</code></pre>
</div>
</div>
<div id="now-you-have-your-count-matrix---you-can-perform-operations-on-sample-groups-to-filter-visualize-data-annotate-your-peaks-and-more" class="section level2">
<h2 class="hasAnchor">
<a href="#now-you-have-your-count-matrix---you-can-perform-operations-on-sample-groups-to-filter-visualize-data-annotate-your-peaks-and-more" class="anchor"></a>Now you have your count matrix - you can perform operations on sample groups to filter, visualize data, annotate your peaks, and more…</h2>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kathryn Rozen-Gagnon, Thomas Carroll, Ji-Dung Luo, Wei Wang, Troels Scheel.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
