<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>CTK: Standard/BrdU-CLIP data analysis • CLIPflexR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="CTK: Standard/BrdU-CLIP data analysis">
<meta property="og:description" content="CLIPflexR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CLIPflexR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.21</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Processing_to_matrix.html">Processing reads, mapping, and building a count matrix</a>
    </li>
    <li>
      <a href="../articles/StandardandBrdU_Processing_CTK.html">CTK: Standard/BrdU-CLIP data analysis</a>
    </li>
    <li>
      <a href="../articles/chimera_vignette.html">Chimera processing</a>
    </li>
    <li>
      <a href="../articles/installCliPR.html">Using and installing the CTK pipeline using CLIPflexR</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/kathrynrozengagnon/CLIPflexR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="StandardandBrdU_Processing_CTK_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>CTK: Standard/BrdU-CLIP data analysis</h1>
                        <h4 data-toc-skip class="author">Kathryn Rozen-Gagnon, Ji-Dung Luo and Thomas Carroll</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/kathrynrozengagnon/CLIPflexR/blob/HEAD/vignettes/StandardandBrdU_Processing_CTK.Rmd" class="external-link"><code>vignettes/StandardandBrdU_Processing_CTK.Rmd</code></a></small>
      <div class="hidden name"><code>StandardandBrdU_Processing_CTK.Rmd</code></div>

    </div>

    
    
<div class="section level3">
<h3 id="this-vignette-shows-how-to-execute-the-ctk-standardbrdu-clip_data_analysis_using_ctk-vignette-using-clipflexr-from-within-r-">This vignette shows how to execute the <a href="https://zhanglab.c2b2.columbia.edu/index.php/Standard/BrdU-CLIP_data_analysis_using_CTK" class="external-link">CTK “Standard/BrdU-CLIP_data_analysis_using_CTK” vignette</a> using CLIPflexR, from within R.<a class="anchor" aria-label="anchor" href="#this-vignette-shows-how-to-execute-the-ctk-standardbrdu-clip_data_analysis_using_ctk-vignette-using-clipflexr-from-within-r-"></a>
</h3>
</div>
<div class="section level2">
<h2 id="download-data">Download Data<a class="anchor" aria-label="anchor" href="#download-data"></a>
</h2>
<p>Download data from <a href="https://www.ncbi.nlm.nih.gov/sra" class="external-link">SRA</a>. <em>SRR110753</em> was selected as an example in this vignette.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">SRAdb</span><span class="op">)</span></span>
<span><span class="va">mySRA</span> <span class="op">&lt;-</span> <span class="st">"SRAmetadb.sqlite"</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">mySRA</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">mySRA</span> <span class="op">&lt;-</span> <span class="fu">getSRAdbFile</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">sra_con</span> <span class="op">&lt;-</span> <span class="fu">dbConnect</span><span class="op">(</span><span class="fu">dbDriver</span><span class="op">(</span><span class="st">"SQLite"</span><span class="op">)</span>, <span class="va">mySRA</span> <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Fox3_Std</span> <span class="op">&lt;-</span> <span class="st">"SRR1107535.fastq.gz"</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">Fox3_Std</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">getFASTQfile</span><span class="op">(</span><span class="st">"SRR1107535"</span>,</span>
<span>               sra_con <span class="op">=</span> <span class="va">sra_con</span>,</span>
<span>               destDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="process-data">Process data<a class="anchor" aria-label="anchor" href="#process-data"></a>
</h2>
<div class="section level3">
<h3 id="quality-filter-reads-wrapping-ctks-fastq_filter-pl">Quality filter reads <strong>(wrapping CTK’s fastq_filter.pl)</strong><a class="anchor" aria-label="anchor" href="#quality-filter-reads-wrapping-ctks-fastq_filter-pl"></a>
</h3>
<p>Filtering low quality read using R-wrapped <strong>fastq_filter.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#fastq_filter.pl" class="external-link">CTK</a>.<br>
Use fastq/gzipped fast files as input. <strong>outFile</strong>: fastq file after filtering low quality reads; <strong>qsFilter</strong>: filtering criteria, “mean:0-29:20” means the average quality of the first 30 bases must be equal or greater than 20.<br><a href="../reference/ctk_fastqFilter.html">CLIPflexR documentation</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Fox3_Std</span> <span class="op">&lt;-</span> <span class="st">"SRR1107535.fastq.gz"</span></span>
<span><span class="va">Fox3_Std_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ctk_fastqFilter.html">ctk_fastqFilter</a></span><span class="op">(</span><span class="va">Fox3_Std</span>,</span>
<span>                                     outFile <span class="op">=</span> <span class="st">"SRR1107535_FF.fastq"</span>,</span>
<span>                                     qsFilter<span class="op">=</span><span class="st">"mean:0-29:20"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="remove-3-linkers-wrapping-fastxs-fastx_clipper">Remove 3’ linkers <strong>(wrapping FASTX’s fastx_clipper)</strong><a class="anchor" aria-label="anchor" href="#remove-3-linkers-wrapping-fastxs-fastx_clipper"></a>
</h3>
<p>Trimming 3’-end adapter sequences using R-wrapped <strong>fastx_clipper</strong> function from <a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html#fastx_clipper_usage" class="external-link">FASTX_toolkit</a>.<br><strong>adaptor</strong>: CLIP this sequence from the 3’-end of reads, “GTGTCAGTCACTTCCAGCGG” (default); <strong>length</strong>: minimum read length, only keep reads with length equal or greater then this threshold, 18 (default).<br><a href="../reference/fastx_clipper.html">CLIPflexR documentation</a>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Fox3_Std_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fastx_clipper.html">fastx_clipper</a></span><span class="op">(</span><span class="va">Fox3_Std_filtered</span>,outFile <span class="op">=</span> <span class="st">"SRR1107535_clipped.fastq"</span>,adaptor <span class="op">=</span> <span class="st">"GTGTCAGTCACTTCCAGCGG"</span>,length <span class="op">=</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="trim-low-quality-nucleotides-and-short-reads-wrapping-fastxs-fastq_quality_trimmer">Trim low quality nucleotides and short reads <strong>(wrapping FASTX’s fastq_quality_trimmer)</strong><a class="anchor" aria-label="anchor" href="#trim-low-quality-nucleotides-and-short-reads-wrapping-fastxs-fastq_quality_trimmer"></a>
</h3>
<p>Discard reads with low quality bases and and short reads using R-wrapped <strong>fastq_quality_trimmer</strong> function from <a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html" class="external-link">FASTX-toolkit</a>. <strong>outfile</strong>: file name and path of the fastq files after filtering; <strong>qualityThreshold</strong>: the minimum quality threshold. <a href="../reference/fastq_quality_trimmer.html">CLIPflexR documentation</a></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Fox3_Std_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fastq_quality_trimmer.html">fastq_quality_trimmer</a></span><span class="op">(</span><span class="va">Fox3_Std_filtered</span>,</span>
<span>                                           outFile <span class="op">=</span> <span class="st">"SRR1107535_clipFiltered.fastq.gz"</span>,qualityThreshold <span class="op">=</span> <span class="fl">5</span>,minimumLength <span class="op">=</span> <span class="fl">20</span> <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="collapse-exact-ctk_fastq2collapse-wrapping-ctkss-ctk_fastq2collapse-pl">Collapse exact ctk_fastq2collapse <strong>(wrapping CTKs’s ctk_fastq2collapse.pl)</strong><a class="anchor" aria-label="anchor" href="#collapse-exact-ctk_fastq2collapse-wrapping-ctkss-ctk_fastq2collapse-pl"></a>
</h3>
<p>Collapse duplicate sequences into unique reads using R-wrapped <strong>fastq2collapse.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#fastq2collapse.pl" class="external-link">CTK</a>.<br><a href="../reference/ctk_fastq2collapse.html">CLIPflexR documentation</a>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Fox3_Std_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ctk_fastq2collapse.html">ctk_fastq2collapse</a></span><span class="op">(</span><span class="va">Fox3_Std_filtered</span>,</span>
<span>                                           outFile <span class="op">=</span> <span class="st">"SRR1107535_collapsed.fastq.gz"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="remove-random-barcodes-wrapping-ctks-stripbarcode-pl">Remove random barcodes <strong>(wrapping CTK’s stripBarcode.pl)</strong><a class="anchor" aria-label="anchor" href="#remove-random-barcodes-wrapping-ctks-stripbarcode-pl"></a>
</h3>
<p>Strip random barcodes with specific length from the reads using R-wrapped <strong>stripBarcode.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#stripBarcode.pl" class="external-link">CTK</a>.<br><strong>linkerlength</strong>: barcode length; <strong>inputFormat</strong>: format of input sequences, fasta (default) or fastq.<br><a href="../reference/ctk_stripBarcode.html">CLIPflexR documentation</a>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Fox3_Std_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ctk_stripBarcode.html">ctk_stripBarcode</a></span><span class="op">(</span><span class="va">Fox3_Std_filtered</span>,linkerlength <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                                      inputFormat <span class="op">=</span> <span class="st">"fasta"</span>,</span>
<span>                                           outFile <span class="op">=</span> <span class="st">"SRR1107535_stripped.fastq.gz"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="map-processed-reads-to-genome">Map processed reads to genome<a class="anchor" aria-label="anchor" href="#map-processed-reads-to-genome"></a>
</h2>
<div class="section level3">
<h3 id="make-reference-data-for-mm10">Make reference data for mm10<a class="anchor" aria-label="anchor" href="#make-reference-data-for-mm10"></a>
</h3>
<p>Fetch reference sequences from <a href="https://www.bioconductor.org/packages/release/bioc/html/BSgenome.html" class="external-link">BSgenome</a>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">BSgenome.Mmusculus.UCSC.mm10</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="va">listOfSeq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">BSgenome.Mmusculus.UCSC.mm10</span><span class="op">)</span>,<span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">BSgenome.Mmusculus.UCSC.mm10</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">listOfSeq</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">BSgenome.Mmusculus.UCSC.mm10</span><span class="op">)</span></span>
<span><span class="va">toWrite</span> <span class="op">&lt;-</span> <span class="fu">DNAStringSet</span><span class="op">(</span><span class="va">listOfSeq</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"Un|random|hap"</span>,<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">listOfSeq</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu">writeXStringSet</span><span class="op">(</span><span class="va">toWrite</span>,file<span class="op">=</span><span class="st">"mm10.fa"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="build-a-bowtie2-genome-index-using-rbowtie2">Build a Bowtie2 genome index using Rbowtie2<a class="anchor" aria-label="anchor" href="#build-a-bowtie2-genome-index-using-rbowtie2"></a>
</h3>
<p>Build Bowtie2 index of reference genome, mm10 in this vignette.<br><strong>threads</strong>: number of threads used. <a href="../reference/bowtie2_index.html">Index documentation</a>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/bowtie2_index.html">bowtie2_index</a></span><span class="op">(</span><span class="st">"mm10.fa"</span>,threads <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="align-to-genome-using-rbowtie2">Align to genome using Rbowtie2<a class="anchor" aria-label="anchor" href="#align-to-genome-using-rbowtie2"></a>
</h3>
<p>Align reads to indexed reference genome using Rbowtie2.<br><strong>format</strong>: format of input sequences, fasta (default) or fastq; <strong>threads</strong>: number of threads used, 1 (default); <strong>maxMismatches</strong>: number of mismatches allowed, 1 (default) or 0; <strong>seedSubString</strong>: length of seed substrings, 18 (default), must be &gt;3 &amp; &lt;32; <strong>report_k</strong>: NULL, look for multiple alignments, report best (default), otherwise report up to n (integer) alignments per read.<br><a href="../reference/bowtie_align.html">Alignment documentation</a>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Fox3_Std_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/decompress.html">decompress</a></span><span class="op">(</span><span class="va">Fox3_Std_filtered</span><span class="op">)</span></span>
<span><span class="va">alignedFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bowtie_align.html">bowtie_align</a></span><span class="op">(</span><span class="va">Fox3_Std_filtered</span>,</span>
<span>              <span class="st">"mm10"</span>,</span>
<span>              <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\.fastq"</span>,<span class="st">".bam"</span>,<span class="va">Fox3_Std_filtered</span><span class="op">)</span>,</span>
<span>              format<span class="op">=</span><span class="st">"fastq"</span>,</span>
<span>              threads <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="parsing-sam-files">Parsing SAM files<a class="anchor" aria-label="anchor" href="#parsing-sam-files"></a>
</h3>
<p>Parsing SAM files to get unique mapped reads and mismatches per read using R-wrapped <strong>parseAlignment.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#parseAlignment.pl" class="external-link">CTK</a>.<br><strong>minLen</strong>: miminal mapping length to report, 18 (default); <strong>mapQual</strong>: Minimum map quality, 1 (default MAPQ), keep only uniquely mapped reads when mapQual&gt;=1.<br><a href="../reference/ctk_parseAlignment.html">CLIPflexR documentation</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">parsedAlignement</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ctk_parseAlignment.html">ctk_parseAlignment</a></span><span class="op">(</span><span class="va">alignedFile</span>,</span>
<span>                                       minLen <span class="op">=</span> <span class="fl">18</span>,</span>
<span>                                       mapQual <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                       mutationFile <span class="op">=</span> <span class="st">"mutations.txt"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="collapse-duplicated-mapped-reads">Collapse duplicated mapped reads<a class="anchor" aria-label="anchor" href="#collapse-duplicated-mapped-reads"></a>
</h3>
<p>Collapse CLIP tags generated by parseAlignment using R-wrapped <strong>tag2collapse.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#tag2collapse.pl" class="external-link">CTK</a>.<br><strong>bigFile</strong>: set TRUE if input file is big; <strong>keepMaxScore</strong>: keep the tag with the most weight (instead of the longest one) as representative; <strong>keepTagName</strong>: do not change tag name (no extra information); <strong>weight</strong>: consider the weight of each tag; <strong>weightInName</strong>: find weight in name; <strong>randomBarcode</strong>: random barcode exists, no collapse for different barcodes; <strong>em</strong>: EM threshold to infer reliability of each collapsed read (when have random linker, -1 = no EM); <strong>outputSeqError</strong>: output sequencing errors estimated by the EM algorithm.<br><a href="../reference/ctk_tag2collapse.html">CLIPflexR documentation</a>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># -v -big --random-barcode -EM 30 --seq-error-model alignment -weight --weight-in-name --keep-max-score --keep-tag-name $f.tag.bed</span></span>
<span><span class="va">tagcollapsed</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ctk_tag2collapse.html">ctk_tag2collapse</a></span><span class="op">(</span><span class="va">parsedAlignement</span>,</span>
<span>                                     keepMaxScore <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                     keepTagName <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                     weight <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                     bigFile <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                     weightInName <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                     randomBarcode <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                     em <span class="op">=</span> <span class="fl">30</span>,</span>
<span>                                     outputSeqError <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>                                     <span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="call-peaks-visualize">Call peaks &amp; visualize<a class="anchor" aria-label="anchor" href="#call-peaks-visualize"></a>
</h2>
<div class="section level3">
<h3 id="join-tags-and-mutations">Join tags and mutations<a class="anchor" aria-label="anchor" href="#join-tags-and-mutations"></a>
</h3>
<p>Get the mutations in unique tags using R-wrapped <strong>selectRow.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#selectRow.pl" class="external-link">CTK</a> (originally wrapped from Galaxy).<br><strong>file1</strong>: mutation file; <strong>file2</strong>: tagged bed file; <strong>field1</strong>: query column; <strong>field2</strong>: filter column; <strong>mode</strong>: running mode, set N while only print paired rows.<br><a href="../reference/ctk_joinWrapper.html">CLIPflexR documentation</a>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tagJoined</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ctk_joinWrapper.html">ctk_joinWrapper</a></span><span class="op">(</span><span class="st">"mutations.txt"</span>,<span class="va">tagcollapsed</span>,field1 <span class="op">=</span> <span class="fl">4</span>,field2 <span class="op">=</span> <span class="fl">4</span>,mode <span class="op">=</span> <span class="st">"n"</span>,</span>
<span>                                outFile<span class="op">=</span><span class="st">"Fox3_3_tag.unique.mutation.txt"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="label-bed-file-with-rgb-information">Label bed file with rgb information<a class="anchor" aria-label="anchor" href="#label-bed-file-with-rgb-information"></a>
</h3>
<p>Add colors to bed file using R-wrapped <strong>bed2rgb.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#bed2rgb.pl" class="external-link">CTK</a>.<br><strong>filesToRun</strong>: taged bed files; <strong>col</strong>: assign color in r,g,b format.<br><a href="../reference/ctk_bed2rgb.html">CLIPflexR documentation</a>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tagcollapsedRGB</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ctk_bed2rgb.html">ctk_bed2rgb</a></span><span class="op">(</span>filesToRun <span class="op">=</span> <span class="va">tagcollapsed</span>,col <span class="op">=</span> <span class="st">"128,0,0"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="generate-bedgraph">Generate bedgraph<a class="anchor" aria-label="anchor" href="#generate-bedgraph"></a>
</h3>
<p>Generate bedgraphs for visualization using R-wrapped <strong>tag2profile.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#tag2profile.pl" class="external-link">CTK</a>.<br><strong>bigFile</strong>: set TRUE if input file is big; <strong>ss</strong>: separate strand, TRUE (default); <strong>exact</strong>: exact count of each nucleotide, TRUE (default).<br><a href="../reference/ctk_tag2profile.html">CLIPflexR documentation</a>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># -v -big --random-barcode -EM 30 --seq-error-model alignment -weight --weight-in-name --keep-max-score --keep-tag-name $f.tag.bed</span></span>
<span><span class="va">tagProfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ctk_tag2profile.html">ctk_tag2profile</a></span><span class="op">(</span>filesToRun <span class="op">=</span> <span class="va">tagcollapsedRGB</span>,bigFile <span class="op">=</span> <span class="cn">TRUE</span>,ss <span class="op">=</span> <span class="cn">TRUE</span>,exact <span class="op">=</span> <span class="cn">TRUE</span>,outputFormat <span class="op">=</span> <span class="st">"bedgraph"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="peak-calling-without-statistical-assessment">Peak calling without statistical assessment<a class="anchor" aria-label="anchor" href="#peak-calling-without-statistical-assessment"></a>
</h3>
<p>Call peaks using R-wrapped <strong>tag2peak.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#tag2peak.pl" class="external-link">CTK</a> <em>without</em> statistical assessment. <strong>filesToRun</strong>: RGB labeled tagged bed files; <strong>valleySeeking</strong>: find candidate peaks by valley seeking, TRUE (default); <strong>valleyDepth</strong>: depth of valley, if valley seeking, 0.9 (default), between 0.5 and 1; <strong>bigFile</strong>: set TRUE if the input file is big; <strong>ss</strong>: separate strand, TRUE (default).<br><a href="../reference/ctk_tag2peak.html">CLIPflexR documentation</a>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tagPeaksNoStats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ctk_tag2peak.html">ctk_tag2peak</a></span><span class="op">(</span>filesToRun <span class="op">=</span> <span class="va">tagcollapsedRGB</span>,valleySeeking <span class="op">=</span> <span class="cn">TRUE</span>,valleyDepth <span class="op">=</span> <span class="fl">0.9</span>,bigFile <span class="op">=</span> <span class="cn">TRUE</span>,ss <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="peak-calling-with-statistical-assessment">Peak calling with statistical assessment<a class="anchor" aria-label="anchor" href="#peak-calling-with-statistical-assessment"></a>
</h3>
<p>Call peaks using R-wrapped <strong>tag2peak.pl</strong> from <a href="https://zhanglab.c2b2.columbia.edu/index.php/CTK_usage#tag2peak.pl" class="external-link">CTK</a> <em>with</em> statistical assessment. <strong>filesToRun</strong>: RGB labeled tagged bed files; <strong>valleySeeking</strong>: find candidate peaks by valley seeking, TRUE (default);<strong>valleyDepth</strong>: depth of valley, if valley seeking, 0.9 (default), between 0.5 and 1; <strong>bigFile</strong>: set TRUE if the input file is big; <strong>ss</strong>: separate strand, TRUE (default); <strong>genes</strong>: gene locus in bed format.<br><a href="../reference/ctk_tag2peak.html">CLIPflexR documentation</a>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">TxDb.Mmusculus.UCSC.mm10.knownGene</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/BEDFile-class.html" class="external-link">export.bed</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">TxDb.Mmusculus.UCSC.mm10.knownGene</span><span class="op">)</span>,<span class="st">"genes.bed"</span><span class="op">)</span></span>
<span><span class="va">tagProfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ctk_tag2peak.html">ctk_tag2peak</a></span><span class="op">(</span>filesToRun <span class="op">=</span> <span class="va">tagcollapsedRGB</span>,</span>
<span>                           outFile <span class="op">=</span> <span class="st">"TC_SRR1107535_stripped.RGB.peakWithStats.bed"</span>,</span>
<span>                           outBoundary <span class="op">=</span><span class="st">"TC_SRR1107535_stripped.RGB.peakBoundary.bed"</span>,</span>
<span>                           outHalfPH <span class="op">=</span> <span class="st">"TC_SRR1107535_stripped.RGB.peakHalf.bed"</span>,</span>
<span>                           valleySeeking <span class="op">=</span> <span class="cn">TRUE</span>,valleyDepth <span class="op">=</span> <span class="fl">0.9</span>,bigFile <span class="op">=</span> <span class="cn">TRUE</span>,ss <span class="op">=</span> <span class="cn">TRUE</span>,genes <span class="op">=</span> <span class="st">"genes.bed"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="use-clipflexr-to-get-peak-sequences-or-search-for-motifs-in-peaks">Use CLIPflexR to get peak sequences or search for motifs in peaks<a class="anchor" aria-label="anchor" href="#use-clipflexr-to-get-peak-sequences-or-search-for-motifs-in-peaks"></a>
</h2>
<div class="section level3">
<h3 id="retrieve-sequence-under-peaks-using-clipflexr-fetchsequencesforclip">Retrieve sequence under peaks using CLIPflexR <strong>fetchSequencesForClIP</strong><a class="anchor" aria-label="anchor" href="#retrieve-sequence-under-peaks-using-clipflexr-fetchsequencesforclip"></a>
</h3>
<p><strong>peaks</strong>: tagged bed file or GRanges of peaks; <strong>reSize</strong>: resize peak to this width around peak center, 64 (default); <strong>fasta</strong>: reference sequence.<br><a href="../reference/fetchSequencesForClIP.html">CLIPflexR documentation</a>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myseq</span> <span class="op">&lt;-</span> <span class="fu">fetchSequencesForClIP</span><span class="op">(</span>peaks <span class="op">=</span> <span class="va">tagProfile</span>,reSize <span class="op">=</span> <span class="fl">20</span>,fasta <span class="op">=</span> <span class="st">"mm10.fa"</span>,bedHeader <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="retrieve-peaks-associated-with-particular-motifs-using-clipflexr-annotatepeakswithpatterns">Retrieve peaks associated with particular motifs using CLIPflexR <strong>annotatePeaksWithPatterns</strong><a class="anchor" aria-label="anchor" href="#retrieve-peaks-associated-with-particular-motifs-using-clipflexr-annotatepeakswithpatterns"></a>
</h3>
<p><strong>peaks</strong>: tagged bed file or GRanges of peaks; <strong>reSize</strong>: resize peak to this width around peak center, 64 (default); <strong>fasta</strong>: reference sequence; <strong>pattern</strong>: FASTA of motif sequence(s).<br><a href="../reference/annotatePeaksWithPatterns.html">CLIPflexR documentation</a>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotatePeaksWithPatterns.html">annotatePeaksWithPatterns</a></span><span class="op">(</span>peaks <span class="op">=</span> <span class="va">tagProfile</span>,patterns <span class="op">=</span> <span class="st">"TGCATG"</span>,resize <span class="op">=</span> <span class="fl">100</span>,fasta <span class="op">=</span> <span class="st">"mm10.fa"</span>,bedHeader<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Kathryn Rozen-Gagnon, Thomas Carroll, Ji-Dung Luo, Wei Wang, Troels Scheel.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
